---
title: "R Notebook"
output: html_notebook
---

# Before you begin:

These scripts were tailored for the analyses performed in: 

Seibert et al, 2022, *Pathobiology and dysbiosis of the respiratory and intestinal microbiota in 14 months old Golden Syrian hamsters infected with SARS-CoV-2*

# Purpose: 
The purpose of this script is to proccess samples to remove any variants with unknown phylas, are classified as Eukaryotes, Chloroplasts or Mitochondria. 

# Load the needed packages
```{r,message=FALSE}
library(phyloseq)
library(dplyr)
library(ggplot2)
library(eulerr)
library(ggpubr)
library(vegan)
library(microbiome)
library(harrietr)
library(phyloseq.extended)
library(metagMisc)
library(data.table)
library(DESeq2)
```

## Import files from dada2

First, Import the ASV count table, taxonomy file and the metadata file.\ 

*In order to read the metadata into a phyloseq object you need to make the SampleID (first column) the row headers*
```{r}
# Set working directory
setwd = "/Users/.."

# Import metadata file 
metadata <- read.csv("/Users/Sars_hamster_metadata.csv", header=T, row.names=1, check.names=F)

# Import count table 
count_i <- read.table("/Users/ASVs_counts_intestine.tsv", header=T, row.names=1, check.names=F, sep="\t")

# Import taxonomy file 
taxa_i <- as.matrix(read.table("/Users/ASVs_taxonomy_dada_wSpecies_intestine.tsv", header=T, row.names=1, check.names=F, sep="\t"))
```

## Filtering sequences for ONLY BACTERIA  
<style>
div.rose { background-color:#ffc4ef; border-radius: 10px; padding: 40px;}
</style>
<div class = "rose">
When the organisms from a set of samples are well-represented in the taxonomic reference database, it is advisable to filter variants in which a high-rank taxonomy could not be assigned. These are most likely sequence artifacts that donâ€™t exist in nature. Therefore, we will filter out any variants that are not classified as bacteria, have an unclassified phyla or are identified as Chloroplast or Mitochondria.
</div>
<br>

**Import files into PhyloSeq**
```{r}
# Import count table into phyloseq
ps_i = otu_table(count_i, taxa_are_rows = TRUE)

# Import taxonomy table into phyloseq 
tax_i = tax_table(taxa_i)

# Import metadata into phyloseq object
sample = sample_data(metadata)

# Merge the count table, taxonomy table and sample data 
Phylo_i = merge_phyloseq(ps_i, tax_i, sample)
Phylo_i
```

**First filter out any Phyla that are unassigned**
```{r}
# Create table, number of features for each phyla
table(tax_table(Phylo_i)[, "Phylum"], exclude = NULL)

# Remove the ASVs and name as uncharacterized if the Phylum has an NA 
Phylo.filter_i <- subset_taxa(Phylo_i, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
table(tax_table(Phylo.filter_i)[, "Phylum"], exclude = NULL)
```

**Next filter out Eukaryotes, chloroplasts and mitochondria, because we only intended to amplify bacterial sequences**
```{r}
# Remove the ASVs and name as uncharacterized if the Phylum has an NA 
table(tax_table(Phylo.filter_i)[, "Kingdom"], exclude = NULL)
Phylo.filter_i
Phylo.filter_filter_i <- Phylo.filter_i %>%
  subset_taxa(
    Kingdom == "Bacteria" &
    Class   != "Chloroplast" &
    Family  != "Mitochondria")
Phylo.filter_filter_i
table(tax_table(Phylo.filter_filter_i)[, "Kingdom"], exclude = NULL)
```

**Separate the samples from the negative and positive controls**
```{r}
# Subset samples to sampleClass in metadata as sample while non-samples include Extraction or PCR 
Phylo.samples_i <- Phylo.filter_filter_i %>%
  subset_samples(sampleClass == "sample")
Phylo.samples_i

Phylo.control_i <- Phylo.filter_filter_i %>%
  subset_samples(sampleClass == "Extraction" | sampleClass == "PCR")
Phylo.control_i
```

**Filter low abundance taxa and visualize the abundance of these taxa in all samples**
```{r}
# Filter OTUs that are found in at least 5% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
Phylo.samples_i_filter <- phyloseq_filter_prevalence(Phylo.samples_i, prev.trh = 0.05, abund.trh = 10, threshold_condition = "AND")
Phylo.samples_i
Phylo.samples_i_filter

# Compute prevalence of each feature, store as data.frame
prevdf = apply(X = otu_table(Phylo.samples_i_filter),
               MARGIN = ifelse(taxa_are_rows(Phylo.samples_i_filter), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})

# Add taxonomy and total read counts to this data.frame
prevdf = data.frame(Prevalence = prevdf,
                    TotalAbundance = taxa_sums(Phylo.samples_i_filter),
                    tax_table(Phylo.samples_i_filter))

# Subset to the remaining phyla
prevdf1 = subset(prevdf, Phylum %in% get_taxa_unique(Phylo.samples_i_filter, "Phylum"))
ggplot(prevdf1, aes(TotalAbundance, Prevalence / nsamples(Phylo.samples_i_filter),color=Phylum)) +
  # Include a guess for parameter
  geom_hline(yintercept = 0.05, alpha = 0.5, linetype = 2) +  geom_point(size = 2, alpha = 0.7) +
  scale_x_log10() +  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_wrap(~Phylum) + theme(legend.position="none")
```

# Save the files 
```{r}
# Save phyloseq files to a R file so that i can reload the filtered file 
saveRDS(object = Phylo.samples_i_filter, file = "/Users/Phylo.samples_i.rds")
```

# Purpose: 
The purpose of the script below is to analyze alpha diversity of the different groups within the intestine and the lungs.

# Load the needed packages
```{r,message=FALSE}
library(vegan)
library(phyloseq)
library(ggplot2)
library(ggpubr)
library(dplyr)
library(microbiome)
```

## Before performing alpha diveristy analysis, the sequences will need to be rarified. 
Rarify the sequences using the vegan package 

### Before rarifying the data in vegan, convert the phyloseq object to a dataframe that is worke-able in vegan. We will do this by creating a function
```{r}
# Create a function to convert the sample_data() within a phyloseq object to a vegan compatible data object
pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

# Create a function to convert the count table within a phyloseq object to a vegan compatible data object
psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

Phylo.samples_i_filter
# Use the function to create a dataframe of the sample data 
sampledataveg_i<- pssd2veg(Phylo.samples_i_filter)
saveRDS(sampledataveg_i, "sampledataveg_i.rds")

# Use the function to create a matrix of the ASV table. This will be a matrix that has ASV number as column headers and sample IDs as row names 
ASVtableveg_i <- psotu2veg(Phylo.samples_i_filter)
```

### Figure 5A: Rarify the SAMPLES in vegan and visualize rarefaction curve
```{r}
# Calculate total OTU counts for all samples (sum all columns)
total.counts <- apply(ASVtableveg_i, 1, sum) 

# List all sample sums in descending order - useful for deciding what your sample cutoff should be
sort(total.counts, decreasing=TRUE)

# List the minimum coverage (sequences per sample)
min(total.counts)

# Change as appropriate for samples. We will not include samples BS-501 since that has a very low coverage of 2,992
sample.size_i <- 27020

# Subset sample.shared to remove samples with < sample.size sequences
sample.used_i <- ASVtableveg_i[total.counts>=sample.size_i,] 

# Resample all samples to a total of sample.size sequences (I do not want to rarify again so i will load sample.rare from rarified data previously analyzed)
sample.rare_i <- rrarefy(sample.used_i, sample=sample.size_i) 

# Save sample.rare to a file so that i can reload the rarified file 
saveRDS(object = sample.rare_i, file = "/Users/RarifiedASVs_i.rds")

sample.rare_i <- readRDS(file = "/Users/RarifiedASVs_i.rds")

library("phyloseq.extended")

rare_SampleType <- ggrare(Phylo.samples_i_filter, step = 500, color = "SampleType")

colorgroups <- c("limegreen", "deeppink2", "darkorange", "dodgerblue3")

# FIGURE 5A
pdf("/Users/RarefactionCurve_Intestine_SampleType.pdf",
    width = 6, height = 3) 

rare_SampleType + scale_color_manual(values = colorgroups, breaks = c('Duodenum', 'Ileum','Cecum', 'Feces'))+
  scale_fill_manual(values = colorgroups, breaks = c('Duodenum', 'Ileum','Cecum', 'Feces'))+
  geom_vline(xintercept = 27020, color = "black", linetype="dashed", size = 0.6)+
  scale_y_continuous(breaks = seq(0, 1700, by=200), limits=c(0, 1700))+
  scale_x_continuous(breaks = seq(0, 80000, by=20000), limits=c(0, 90000))+
  theme_bw()+
  theme(panel.grid.minor = element_blank())+
  theme(axis.title.x=element_text(colour = "black"),
          axis.title.y = element_text(colour = "black"),
          axis.text.x = element_text(colour = "black", size = 12),
          axis.text.y = element_text(colour = "black", size = 12),
          axis.line.x = element_line(color="black"),
          axis.line.y = element_line(color="black"),
          legend.text = element_text(color="black"))+
  theme(axis.line = element_line(color="black", size = 0.2),
          panel.border = element_blank())

dev.off()
```

## Alpha diversity of samples separated by group combining all dpc 

**Alpha diversity**
<style>
div.rose { background-color:#ffc4ef; border-radius: 10px; padding: 40px;}
</style>
<div class = "rose">
Estimating alpha diversity of microbial communities is problematic no matter what you do. The best is to subsample the libraries with replacement to estimate the species abundance of the real population while standardizing sampling effort.
</div>
<br>

### Import the rarified data into phyloseq since that is how we are going to calculate the alpha diversity metrics 
```{r}
# Import rarified count into phyloseq
psrare_i <- otu_table(sample.rare_i, taxa_are_rows = FALSE)

# Import taxonomy table into phyloseq 
tax_i = tax_table(taxa_i)

# Import the sample metadata into phyloseq object
sample = sample_data(metadata)

# Merge the count table, taxonomy table and sample data 
Phylo.rare_i  = merge_phyloseq(psrare_i, tax_i, sample)
Phylo.rare_i

# Save sample.rare to a file so that i can reload the rarified file 
saveRDS(object = Phylo.rare_i, file = "/Users/Phylo.rare_i.rds")
```

### Calculate the alpha diversity metrics and plot the graphs in ggplot 
```{r}
# Import metadata file 2 with sampled ID as a . instead of a -
metadata_2 <- read.csv("/Users/Sars_hamster_metadata_2.csv", header=T, row.names=1, check.names=F)

# Calculate diversity metrics in phyloseq 
PhlyoRichness_i <- estimate_richness(Phylo.rare_i, measures = c("Observed", "Shannon", "InvSimpson", "Fisher"))

# Make a column that contains the rownames so that we have a column of the SampleIDs
PhlyoRichness_i$SampleID <- rownames(PhlyoRichness_i)

# Join the diversity tables with metadata 
PhlyoRichness_meta_i = left_join(PhlyoRichness_i, metadata_2, by = "SampleID")

# Re-level the groups 
PhlyoRichness_meta_i$Group_PostInfect<- factor(PhlyoRichness_meta_i$Group_PostInfect, levels = c('SARS-CoV-2', 'IAV-SARS-CoV-2', 'Mock', 'Pre-Challenge'))

PhlyoRichness_meta_i$SampleType<- factor(PhlyoRichness_meta_i$SampleType, levels = c('Duodenum','Ileum', 'Cecum', 'Feces'))

# Export data 
write.csv(PhlyoRichness_meta_i, "/Users/AlphaDiversityIndexes_intestinefeces.csv")

# Set the colors for all of the graphs 
colorgroups = c("coral2", "dodgerblue1", "white", "grey66")
```

### Figure 5B: Next, visualize the number of observed ASVs in the different groups
```{r}
# Filter so that you only have observed values
PhlyoRichness_meta_i.observed <- PhlyoRichness_meta_i[,c(1,7,8,9, 10,11)]

# FIGURE 5B
ggplot(PhlyoRichness_meta_i.observed, aes(x=SampleType, y=Observed, fill=Group_PostInfect, ))+ 
  geom_boxplot()+
  geom_point(shape = 21, 
             size = 3,
             colour = "Black",
             show.legend = F,
             alpha = 0.3, 
             position = position_dodge(0.75))+
  scale_fill_manual(values = colorgroups)+
  scale_y_continuous(breaks = seq(0, 1500, by=250), limits=c(0, 1500))+
  ylab("Observed ASVs")+
  theme_gray()+
  guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(vjust=1.5, size = 20),
        axis.text.x = element_text(colour = "black", size = 22),
        axis.text.y = element_text(colour = "black", size = 22),
        text = element_text(size=24),
        strip.text.y = element_text(angle = 0),
        legend.title = element_text(size = 16, face = "bold"), 
        legend.text = element_text(size = 14))+
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5),
          panel.border = element_blank(),
          panel.background = element_blank())
  #stat_compare_means(method = "kruskal.test")

# Statistics testing for comparison among means of the bray curtis for different groups (this uses the Wilcoxon Sign test (non-parametric))

# Filter for only duodenum and then perform pairwise comparison 
PhlyoRichness_meta_i.observed_Duodenum <- PhlyoRichness_meta_i.observed %>%
  filter(SampleType == "Duodenum")
compare_means(Observed ~ Group_PostInfect,  data = PhlyoRichness_meta_i.observed_Duodenum, method = "wilcox.test", p.adj = "bonferroni")

# Filter for only ileum and then perform pairwise comparison 
PhlyoRichness_meta_i.observed_Ileum <- PhlyoRichness_meta_i.observed %>%
  filter(SampleType == "Ileum")
compare_means(Observed ~ Group_PostInfect,  data = PhlyoRichness_meta_i.observed_Ileum, method = "wilcox.test", p.adj = "bonferroni")

# Filter for only cecum and then perform pairwise comparison 
PhlyoRichness_meta_i.observed_Cecum <- PhlyoRichness_meta_i.observed %>%
  filter(SampleType == "Cecum")
compare_means(Observed ~ Group_PostInfect,  data = PhlyoRichness_meta_i.observed_Cecum, method = "wilcox.test", p.adj = "bonferroni")

# Filter for only feces and then perform pairwise comparison 
PhlyoRichness_meta_i.observed_Feces <- PhlyoRichness_meta_i.observed %>%
  filter(SampleType == "Feces")
compare_means(Observed ~ Group_PostInfect,  data = PhlyoRichness_meta_i.observed_Feces, method = "wilcox.test", p.adj = "bonferroni")

```

### Figure S5B: Calculate the alpha diversity metrics of pre-challenge fecal samples with and without FLUAV exposure
```{r}
# Calculate diversity metrics in phyloseq 
PhlyoRichness_i <- estimate_richness(Phylo.rare_i, measures = c("Observed", "Shannon", "InvSimpson", "Fisher"))

# Make a column that contains the rownames so that we have a column of the SampleIDs
PhlyoRichness_i$SampleID <- rownames(PhlyoRichness_i)

# Join the diversity tables with metadata 
PhlyoRichness_meta_i = left_join(PhlyoRichness_i, metadata_2, by = "SampleID")

PhlyoRichness_meta_i_feces <- PhlyoRichness_meta_i %>%
  filter(SampleType == "Feces") %>%
  filter(Group_PostInfect == "Pre-Challenge")


# Classify mock and pre-challenge
PhlyoRichness_meta_i_feces <- PhlyoRichness_meta_i_feces %>% mutate(GroupPreChallenge =
                     case_when(HamsterID == "1" ~ "Mock", 
                               HamsterID == "2" ~ "Mock",
                               HamsterID == "3" ~ "Mock",
                               HamsterID == "4" ~ "Mock",
                               HamsterID == "5" ~ "Mock",
                               HamsterID == "7" ~ "Mock",
                               HamsterID == "8" ~ "Mock",
                               HamsterID == "9" ~ "Mock",
                               HamsterID == "10" ~ "Mock",
                               HamsterID == "11" ~ "FLUAV",
                               HamsterID == "12" ~ "FLUAV",
                               HamsterID == "13" ~ "FLUAV",
                               HamsterID == "14" ~ "FLUAV",
                               HamsterID == "15" ~ "FLUAV",
                               HamsterID == "16" ~ "FLUAV",
                               HamsterID == "17" ~ "FLUAV")
)

# Set the colors for all of the graphs 
colorgroups = c("green4", "purple")

# Re-level the groups 
PhlyoRichness_meta_i_feces$GroupPreChallenge<- factor(PhlyoRichness_meta_i_feces$GroupPreChallenge, levels = c('FLUAV','Mock'))


# FIGURE S5B
ggplot(PhlyoRichness_meta_i_feces, aes(x=GroupPreChallenge, y=Observed, fill=GroupPreChallenge,))+ 
  geom_boxplot()+
  geom_point(shape = 21, 
             size = 3,
             colour = "Black",
             show.legend = F,
             alpha = 0.3, 
             position = position_dodge(0.75))+
  scale_fill_manual(values = colorgroups)+
  scale_y_continuous(breaks = seq(0, 1500, by=250), limits=c(0, 1500))+
  ylab("Observed ASVs")+
  theme_gray()+
  guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(vjust=1.5, size = 20),
        axis.text.x = element_text(colour = "black"),
        axis.text.y = element_text(colour = "black"),
        text = element_text(size=24),
        strip.text.y = element_text(angle = 0),
        legend.title = element_text(size = 16, face = "bold"), 
        legend.text = element_text(size = 14))+
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5),
          panel.border = element_blank(),
          panel.background = element_blank())+
  stat_compare_means(method = "wilcox.test")

ggplot(PhlyoRichness_meta_i_feces, aes(x=GroupPreChallenge, y=Shannon, fill=GroupPreChallenge, ))+ 
  geom_boxplot()+
  geom_point(shape = 21, 
             size = 3,
             colour = "Black",
             show.legend = F,
             alpha = 0.3, 
             position = position_dodge(0.75))+
  scale_fill_manual(values = colorgroups)+
  scale_y_continuous(breaks = seq(0, 8, by=2), limits=c(0, 8))+
  ylab("Shannon")+
  theme_gray()+
  guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(vjust=1.5, size = 20),
        axis.text.x = element_text(colour = "black"),
        axis.text.y = element_text(colour = "black"),
        text = element_text(size=24),
        strip.text.y = element_text(angle = 0),
        legend.title = element_text(size = 16, face = "bold"), 
        legend.text = element_text(size = 14))+
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5),
          panel.border = element_blank(),
          panel.background = element_blank())+
  stat_compare_means(method = "wilcox.test")
```

### Figure S5A: Next, graph shannon index among the different groups
```{r}
# Filter so that you only have observed values
PhlyoRichness_meta_i.shannon <- PhlyoRichness_meta_i[,c(2,7,8,9, 10,11)]

PhlyoRichness_meta_i.shannon$Group_PostInfect<- factor(PhlyoRichness_meta_i.shannon$Group_PostInfect, levels = c('SARS-CoV-2','IAV-SARS-CoV-2', 'Mock', 'Pre-Challenge'))

PhlyoRichness_meta_i.shannon$SampleType<- factor(PhlyoRichness_meta_i.shannon$SampleType, levels = c('Duodenum','Ileum', 'Cecum', 'Feces'))

# Set the colors for all of the graphs 
colorgroups = c("coral2", "dodgerblue1", "white", "grey66")

# FIGURE S5A
ggplot(PhlyoRichness_meta_i.shannon, aes(x=SampleType, y=Shannon, fill=Group_PostInfect, ))+ 
  geom_boxplot()+
  geom_point(shape = 21, 
             size = 3,
             colour = "Black",
             show.legend = F,
             alpha = 0.3, 
             position = position_dodge(0.75))+
  scale_fill_manual(values = colorgroups)+
  scale_y_continuous(breaks = seq(0, 6, by=2), limits=c(0, 6))+
  ylab("Shannon")+
  theme_gray()+
  guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(vjust=1.5, size = 20),
        axis.text.x = element_text(colour = "black"),
        axis.text.y = element_text(colour = "black"),
        text = element_text(size=24),
        strip.text.y = element_text(angle = 0),
        legend.title = element_text(size = 16, face = "bold"), 
        legend.text = element_text(size = 14))+
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5),
          panel.border = element_blank(),
          panel.background = element_blank())+
  stat_compare_means(method = "kruskal.test")
  
# Statistics testing for comparison among means of the bray curtis for different groups (this uses the Wilcoxon Sign test (non-parametric))

# Filter for only duodenum and then perform pairwise comparison 
PhlyoRichness_meta_i.shannon_Duodenum <- PhlyoRichness_meta_i.shannon %>%
  filter(SampleType == "Duodenum")
compare_means(Shannon ~ Group_PostInfect,  data = PhlyoRichness_meta_i.shannon_Duodenum, method = "wilcox.test", p.adj = "bonferroni")

# Filter for only ileum and then perform pairwise comparison 
PhlyoRichness_meta_i.shannon_Ileum <- PhlyoRichness_meta_i.shannon %>%
  filter(SampleType == "Ileum")
compare_means(Shannon ~ Group_PostInfect,  data = PhlyoRichness_meta_i.shannon_Ileum, method = "wilcox.test", p.adj = "bonferroni")

# Filter for only cecum and then perform pairwise comparison 
PhlyoRichness_meta_i.shannon_Cecum <- PhlyoRichness_meta_i.shannon %>%
  filter(SampleType == "Cecum")
compare_means(Shannon ~ Group_PostInfect,  data = PhlyoRichness_meta_i.shannon_Cecum, method = "wilcox.test", p.adj = "bonferroni")

# Filter for only cecum and then perform pairwise comparison 
PhlyoRichness_meta_i.shannon_Feces <- PhlyoRichness_meta_i.shannon %>%
  filter(SampleType == "Feces")
compare_means(Shannon ~ Group_PostInfect,  data = PhlyoRichness_meta_i.shannon_Feces, method = "wilcox.test", p.adj = "bonferroni")
```

# Beta Diversity 

## Figure S6A AND S6B: Visualize Jaccard (S6A) Bray-Curtis (S6B) distances of the duodenum and ileum together
```{r}
#List the Sample ID numbers that correspond to the duodenum and ileum only
include_list <- c("BS-492", "BS-493", "BS-494", "BS-495", "BS-496", "BS-497", "BS-498", "BS-499", "BS-500", "BS-502", "BS-503", "BS-504", "BS-505", "BS-506", "BS-507", "BS-508", "BS-509", "BS-510", "BS-511", "BS-512", "BS-513", "BS-514", "BS-515", "BS-516", "BS-517", "BS-518", "BS-519", "BS-520", "BS-521", "BS-522", "BS-523")

ASVtableveg_i_SI <- sample.rare_i[rownames(sample.rare_i) %in% include_list, ]

# Calculating relative abundance and creating new dataframe with relative abundance data
ASVtableveg_i.rel <-         
  decostand(ASVtableveg_i_SI, method = "total")

# FIGURE S6A
# Calculate the Jaccard distances (FIGURE S6A)
# The Wisconsin transformation normalizes 0-1 so it will be very similar to the unweighted so i will not use the transformation 
intestine_bray_nmds <-  metaMDS(ASVtableveg_i.rel, 
        distance="jaccard",         # set the diversity metric used
        binary = TRUE,  #https://stats.stackexchange.com/questions/242110/nmds-from-jaccard-and-bray-curtis-identical-is-that-a-bad-thing
        k=3,                     # set number of dimensions
        maxit = 999, 
        trymax=500, 
        autotransform=FALSE)

# Determine the stress value on the number of dimensions
intestine_bray_nmds

# Create a stress plot to visualize the data 
stressplot(intestine_bray_nmds)

# Plot the bray-curtis distances in simple plot 
ordiplot(intestine_bray_nmds, 
         disp="sites")

# Sites: "BS-493" "BS-494" 
#"BS-495" "BS-496" "BS-497" "BS-498" "BS-499" "BS-500" 
#"BS-502" "BS-503" "BS-505" "BS-506" "BS-507" 
#"BS-508" "BS-509" "BS-510" 
#"BS-511" "BS-512" "BS-513" "BS-514" "BS-515" 
#"BS-517" "BS-518" "BS-519" "BS-520" "BS-521" "BS-522" "BS-523" 
#"BS-492" 
#"BS-504" 
#"BS-516"

MyMeta <- c("Mock","Mock",
            "SARS2","SARS2","SARS2","SARS2","SARS2","SARS2",
            "FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2",
            "Mock","Mock","Mock",
            "SARS2","SARS2","SARS2","SARS2","SARS2",
            "FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2",
            "Mock", 
            "FLUAV-SARS2", 
            "SARS2")

colors <- c("black","black",
            "coral2","coral2","coral2","coral2","coral2","coral2",
            "dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1",
            "black","black","black",
            "coral2", "coral2", "coral2", "coral2", "coral2", 
            "dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1",
            "black",
            "dodgerblue1",
            "coral2")

shape <- c(15,19,
           15,15,15,19,19,19,
           15,15,19,19,19,
           15,15,19,
           15,15,15,19,19,
           15,15,15,15,19,19,19,
           15,
           15,
           19) 

plot(intestine_bray_nmds, 
         disp="sites")
ordiellipse(intestine_bray_nmds, MyMeta, kind = "sd", label = T, col = c("dodgerblue1", "black", "coral2"))
#ordilabel(intestine_bray_nmds, disp="sites")
points(intestine_bray_nmds, disp="sites", pch=shape, col = colors, cex = 1)

# FIGURE S6B
# Calculate the Bray-Curtis distances (FIGURE S6A)
intestine_bray_nmds <-  metaMDS(ASVtableveg_i.rel, 
        distance="bray",         # set the diversity metric used
        k=3,                     # set number of dimensions
        maxit = 999, 
        trymax=500, 
        autotransform=FALSE)

# Determine the stress value on the number of dimensions
intestine_bray_nmds

# Create a stress plot to visualize the data 
stressplot(intestine_bray_nmds)

# Plot the bray-curtis distances in simple plot 
ordiplot(intestine_bray_nmds, 
         disp="sites")

plot(intestine_bray_nmds, 
         disp="sites")
ordiellipse(intestine_bray_nmds, MyMeta, kind = "sd", label = T, col = c("dodgerblue1", "black", "coral2"))
#ordilabel(intestine_bray_nmds, disp="sites")
points(intestine_bray_nmds, disp="sites", pch=shape, col = colors, cex = 1)
```

### Perform PERMANOVA analysis to statistically analyze if there are differences among the different groups within the small intestine
```{r}
# PERMANOVA ANALYSIS 
Phylo.samples_i_SI <- subset_samples(Phylo.rare_i, SampleType=="Duodenum" | SampleType == "Ileum")
pseq.rel <- microbiome::transform(Phylo.samples_i_SI, "compositional")
otu <- abundances(pseq.rel)
meta <- meta(pseq.rel)

permanova <- adonis(t(otu) ~ Group_PostInfect*dpi,
               data = meta, permutations=1000, method = "jaccard", by = "terms")
permanova

permanova <- adonis(t(otu) ~ Group_PostInfect*dpi,
               data = meta, permutations=1000, method = "bray", by = "terms")
permanova
```

## Figure S6C and S6D: Visualize Jaccard (S6C) Bray-Curtis (S6D) distances of the cecum using unweighted jaccard and weighted bray-curtis
```{r}
include_list <- c("BS-524", "BS-525", "BS-526", "BS-527", "BS-528", "BS-529", "BS-530", "BS-531", "BS-532", "BS-533", "BS-534", "BS-535", "BS-536", "BS-537", "BS-538", "BS-539")
ASVtableveg_i_Cecum <- sample.rare_i[rownames(sample.rare_i) %in% include_list, ]

# Calculating relative abundance and creating new dataframe with relative abundance data
ASVtableveg_i.rel <-         
  decostand(ASVtableveg_i_Cecum, method = "total")

#FIGURE S6C
# Calculate the Jaccard distances 
# The Wisconsin transformation normalizes 0-1 so it will be very similar to the unweighted so i will not use the transformation 
intestine_bray_nmds <-  metaMDS(ASVtableveg_i.rel, 
        distance="jaccard",         # set the diversity metric used
        binary = TRUE,  #https://stats.stackexchange.com/questions/242110/nmds-from-jaccard-and-bray-curtis-identical-is-that-a-bad-thing
        k=3,                     # set number of dimensions
        maxit = 999, 
        trymax=500, 
        autotransform=FALSE)

# Determine the stress value on the number of dimensions
intestine_bray_nmds

# Create a stress plot to visualize the data 
stressplot(intestine_bray_nmds)

# Plot the bray-curtis distances in simple plot 
ordiplot(intestine_bray_nmds, 
         disp="sites")

# Sites: "BS-524" "BS-525" "BS-526" 
#"BS-527" "BS-529" "BS-530" "BS-531" 
#"BS-533" "BS-534" "BS-535" "BS-536" "BS-537" "BS-538" "BS-539" 
#"BS-528" "BS-532"

MyMeta <- c("Mock","Mock","Mock",
            "SARS2","SARS2","SARS2","SARS2",
            "FLUAV-SARS2", "FLUAV-SARS2", "FLUAV-SARS2", "FLUAV-SARS2", "FLUAV-SARS2", "FLUAV-SARS2", "FLUAV-SARS2", 
            "SARS2", "SARS2")

colors <- c("black","black", "black",
            "coral2","coral2","coral2","coral2",
            "dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1",
            "coral2", "coral2")

shape <- c(15,15,19,
           15,15,19,19,
           15,15,15,15,19,19,19,
           15,19) 

plot(intestine_bray_nmds, 
         disp="sites")
ordiellipse(intestine_bray_nmds, MyMeta, kind = "sd", label = T, col = c("dodgerblue1", "black", "coral2"))
#ordilabel(intestine_bray_nmds, disp="sites")
points(intestine_bray_nmds, disp="sites", pch=shape, col = colors, cex = 1)

# FIGURE S6D
# Calculate the bray-curtis distances 
# The Wisconsin transformation normalizes 0-1 so it will be very similar to the unweighted so i will not use the transformation 
intestine_bray_nmds <-  metaMDS(ASVtableveg_i.rel, 
          distance="bray",         # set the diversity metric used
          k=3,                     # set number of dimensions
          maxit = 999, 
          trymax=500, 
          autotransform=FALSE)

# Determine the stress value on the number of dimensions
intestine_bray_nmds

# Create a stress plot to visualize the data 
stressplot(intestine_bray_nmds)

# Plot the bray-curtis distances in simple plot 
ordiplot(intestine_bray_nmds, 
         disp="sites")

plot(intestine_bray_nmds, 
         disp="sites")
ordiellipse(intestine_bray_nmds, MyMeta, kind = "sd", label = T, col = c("dodgerblue1", "black", "coral2"))
#ordilabel(intestine_bray_nmds, disp="sites")
points(intestine_bray_nmds, disp="sites", pch=shape, col = colors, cex = 1)
```

### Perform PERMANOVA analysis to statistically analyze if there are differences among the different groups within the cecum
```{r}
# PERMANOVA ANALYSIS 
Phylo.samples_i_cecum <- subset_samples(Phylo.rare_i, SampleType=="Cecum")
pseq.rel <- microbiome::transform(Phylo.samples_i_cecum, "compositional")
otu <- abundances(pseq.rel)
meta <- meta(pseq.rel)

permanova <- adonis(t(otu) ~ Group_PostInfect*dpi,
               data = meta, permutations=1000, method = "jaccard", by = "terms")
permanova

permanova <- adonis(t(otu) ~ Group_PostInfect*dpi,
               data = meta, permutations=1000, method = "bray", by = "terms")
permanova
```

## Figure S6E and S6F: Visualize Jaccard (S6E) and Bray-Curtis (S6F) distances of the different groups in the feces
```{r}
include_list <- c("BS-540", "BS-541", "BS-542", "BS-543", "BS-545", "BS-546", "BS-547", "BS-548", "BS-549", "BS-550", "BS-551", "BS-552", "BS-553", "BS-554", "BS-556", "BS-557", "BS-558", "BS-559", "BS-560", "BS-561", "BS-562", "BS-563", "BS-564", "BS-565", "BS-566", "BS-567", "BS-568", "BS-569", "BS-570", "BS-571", "BS-572", "BS-573", "BS-574", "BS-575", "BS-576", "BS-577", "BS-578", "BS-579", "BS-580")
ASVtableveg_i_Feces <- sample.rare_i[rownames(sample.rare_i) %in% include_list, ]

# Calculating relative abundance and creating new dataframe with relative abundance data
ASVtableveg_i.rel <-         
  decostand(ASVtableveg_i_Feces, method = "total")

# FIGURE S6E
# Calculate the Jaccard distances 
# The Wisconsin transformation normalizes 0-1 so it will be very similar to the unweighted so i will not use the transformation 
intestine_bray_nmds <-  metaMDS(ASVtableveg_i.rel, 
        distance="jaccard",         # set the diversity metric used
        binary = TRUE,  #https://stats.stackexchange.com/questions/242110/nmds-from-jaccard-and-bray-curtis-identical-is-that-a-bad-thing
        k=3,                     # set number of dimensions
        maxit = 999, 
        trymax=500, 
        autotransform=FALSE)

# Determine the stress value on the number of dimensions
intestine_bray_nmds

# Create a stress plot to visualize the data 
stressplot(intestine_bray_nmds)

# Plot the bray-curtis distances in simple plot 
ordiplot(intestine_bray_nmds, 
         disp="sites")

# Sites: "BS-540" "BS-541" "BS-542" "BS-543" "BS-545" "BS-546" "BS-547" "BS-548" "BS-549" "BS-550" "BS-551" "BS-552" "BS-553" "BS-554" 
#"BS-557" "BS-558" 
#"BS-559" "BS-560" #"BS-561""BS-562" "BS-563" "BS-564" 
#"BS-565" "BS-566" "BS-567" "BS-568" "BS-569" "BS-570" "BS-571" 
#"BS-572" "BS-573" 
#"BS-574" "BS-575" "BS-576" 
#"BS-577" "BS-578" "BS-579" 
#"BS-556"
#"BS-580"

MyMeta <- c("Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", "Pre-Challenge", 
           "Mock","Mock",
           "SARS2","SARS2","SARS2","SARS2","SARS2","SARS2",
           "FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2",
           "Mock","Mock",
           "SARS2","SARS2","SARS2",
           "FLUAV-SARS2","FLUAV-SARS2","FLUAV-SARS2",
           "Mock",
           "FLUAV-SARS2")

colors <- c("grey66", "grey66","grey66","grey66","grey66","grey66","grey66","grey66","grey66","grey66","grey66","grey66","grey66","grey66",
            "black", "black",
            "coral2","coral2","coral2","coral2","coral2","coral2",
            "dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1",
            "black","black",
            "coral2","coral2","coral2",
            "dodgerblue1","dodgerblue1","dodgerblue1",
            "black",
            "dodgerblue1")

shape <- c(17, 17,17,17,17,17,17,17,17,17,17,17,17,17,
           15, 15, 
           15,15,15,15,15,15,
           15,15,15,15,15,15,15, 
           19, 19, 
           19, 19, 19, 
           19, 19, 19,
           15,
           19) 

plot(intestine_bray_nmds, 
         disp="sites")
ordiellipse(intestine_bray_nmds, MyMeta, kind = "sd", label = T, col = c("dodgerblue1", "black", "grey66","coral2"))
#ordilabel(intestine_bray_nmds, disp="sites")
points(intestine_bray_nmds, disp="sites", pch=shape, col = colors, cex = 1)

# FIGURE S6F
# Calculate the bray-curtis distances 
intestine_bray_nmds <-  metaMDS(ASVtableveg_i.rel, 
          distance="bray",         # set the diversity metric used
          k=3,                     # set number of dimensions
          maxit = 999, 
          trymax=500, 
          autotransform=FALSE)

# Determine the stress value on the number of dimensions
intestine_bray_nmds

# Create a stress plot to visualize the data 
stressplot(intestine_bray_nmds)

# Plot the bray-curtis distances in simple plot 
ordiplot(intestine_bray_nmds, 
         disp="sites")

plot(intestine_bray_nmds, 
         disp="sites")
ordiellipse(intestine_bray_nmds, MyMeta, kind = "sd", label = T, col = c("dodgerblue1", "black", "grey66","coral2"))
#ordilabel(intestine_bray_nmds, disp="sites")
points(intestine_bray_nmds, disp="sites", pch=shape, col = colors, cex = 1)
```

### Perform PERMANOVA analysis to statistically analyze if there are differences among the different groups within the cecum
```{r}
# PERMANOVA ANALYSIS 
Phylo.samples_i_feces <- subset_samples(Phylo.rare_i, SampleType=="Feces")
pseq.rel <- microbiome::transform(Phylo.samples_i_feces, "compositional")
otu <- abundances(pseq.rel)
meta <- meta(pseq.rel)

permanova <- adonis(t(otu) ~ Group_PostInfect*dpi,
               data = meta, permutations=1000, method = "jaccard", by = "terms")
permanova

permanova <- adonis(t(otu) ~ Group_PostInfect*dpi,
               data = meta, permutations=1000, method = "bray", by = "terms")
permanova
```

### Figure S7A: Calculate the jaccard distances and produce a boxplot for the small intestine
```{r}
include_list <- c("BS-492", "BS-493", "BS-494", "BS-495", "BS-496", "BS-497", "BS-498", "BS-499", "BS-500", "BS-501", "BS-502", "BS-503", "BS-504", "BS-505", "BS-506", "BS-507", "BS-508", "BS-509", "BS-510", "BS-511", "BS-512", "BS-513", "BS-514", "BS-515", "BS-516", "BS-517", "BS-518", "BS-519", "BS-520", "BS-521", "BS-522", "BS-523")

ASVtableveg_i_SI <- sample.rare_i[rownames(sample.rare_i) %in% include_list, ]

# Calculating relative abundance and creating new dataframe with relative abundance data
ASVtableveg_i.rel <-         
  decostand(ASVtableveg_i_SI, method = "total")

# Calculate the bray-curtis distances of the duodenum using vegan 
bray_veg_intestine <- vegdist(ASVtableveg_i.rel, 
                          method = "jaccard", 
                          binary = TRUE)

# Turn the distance class into a matrix  
bray_veg_intestine_matrix <- as.matrix(bray_veg_intestine)

# Use program harrietr to melt the distance matrix into 3 column data frame
bray_veg_intestine_df <- melt_dist(bray_veg_intestine_matrix)

# Convert the Var 1 and Var2 to a character 
bray_veg_intestine.melt2 = bray_veg_intestine_df %>%
  filter(as.character(iso1) != as.character(iso2)) %>%
  mutate_if(is.factor, as.character)

# Only keep columns SampleID and GroupGen
sd1 = metadata %>%
  dplyr::select("SampleID", "Group_PostInfect") %>%
  mutate_if(is.factor,as.character)

# Convert the column names to Var1 and Type 1 in metadata sheet sd
colnames(sd1) = c("iso1", "Group1")

# Join the distance matrix and the metadata sheet sd
bray_veg_intestine.melt2.sd = left_join(bray_veg_intestine.melt2, sd1, by = "iso1")

# Convert the column names to Var2 and Type 2 in metadata sheet sd
colnames(sd1) = c("iso2", "Group2")

# Join the distance matrix and the metadata sheet sd
bray_veg_intestine.melt2.sd = left_join(bray_veg_intestine.melt2.sd, sd1, by = "iso2")

write.csv(bray_veg_intestine.melt2.sd, "/Users/jaccard_smallIntestine.csv")

# The spreadsheet was edited to include 2 columns: group1 and then the group 2. The matrix was converted into a dataframe. 
bray_veg_intestine.melt2.sd_edited <- read.csv("/Users/jaccard_smallIntestine_edited.csv", 
                     header=T, 
                     row.names=1, 
                     check.names=F)

colorgroups = c("coral2", "dodgerblue1","black","darkgoldenrod3", "darkgoldenrod3", "darkgoldenrod3")

# Reorder the groups so that they are in the same order
bray_veg_intestine.melt2.sd_edited$Name<- factor(bray_veg_intestine.melt2.sd_edited$Name, levels =c('SARS2', 'FLUAV-SARS2', 'Mock', 'SARS2 v. Mock', 'FLUAV-SARS2 v. Mock','SARS2 v. FLUAV-SARS2'))

# FIGURE S7A
pdf("/Users/Boxplot_Jaccard_SI.pdf",
    width = 11, height = 8) 

ggplot(bray_veg_intestine.melt2.sd_edited, aes(x = Name, y = dist, color = Name)) + 
    geom_boxplot() +
    scale_color_manual(values = colorgroups)+
    scale_y_continuous(breaks = seq(0.2, 1, by=.2), limits=c(0.1, 1.1))+
    ylab("Jaccard Distance")+
    guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
    #theme_bw() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_text(vjust=1.5, size = 22),
          axis.text.x = element_text(angle=40, hjust=1, colour = "black"),
          axis.text.y = element_text(colour = "black"),
          text = element_text(size=25),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5),
          legend.position = "none")+
    theme(panel.grid = element_blank(),
            axis.line = element_line(color="black", size = 0.5),
            panel.border = element_blank(),
            panel.background =  element_rect(fill='white'))

dev.off()

stats <- compare_means(dist ~ Name,  data = bray_veg_intestine.melt2.sd_edited, method = "wilcox.test", p.adj = "bonferroni")  
stats
```

### Figure S7B: Calculate the jaccard distances and produce a boxplot for the cecum
```{r}
include_list <- c("BS-524", "BS-525", "BS-526", "BS-527", "BS-528", "BS-529", "BS-530", "BS-531", "BS-532", "BS-533", "BS-534", "BS-535", "BS-536", "BS-537", "BS-538", "BS-539")
ASVtableveg_i_Cecum <- sample.rare_i[rownames(sample.rare_i) %in% include_list, ]

# Calculating relative abundance and creating new dataframe with relative abundance data
ASVtableveg_i.rel <-         
  decostand(ASVtableveg_i_Cecum, method = "total")

# Calculate the bray-curtis distances of the duodenum using vegan 
bray_veg_intestine <- vegdist(ASVtableveg_i.rel, 
                          method = "jaccard", 
                          binary = TRUE)

# Turn the distance class into a matrix  
bray_veg_intestine_matrix <- as.matrix(bray_veg_intestine)

# Use program harrietr to melt the distance matrix into 3 column data frame
bray_veg_intestine_df <- melt_dist(bray_veg_intestine_matrix)

# Convert the Var 1 and Var2 to a character 
bray_veg_intestine.melt2 = bray_veg_intestine_df %>%
  filter(as.character(iso1) != as.character(iso2)) %>%
  mutate_if(is.factor, as.character)

# Only keep columns SampleID and GroupGen
sd1 = metadata %>%
  dplyr::select("SampleID", "Group_PostInfect") %>%
  mutate_if(is.factor,as.character)

# Convert the column names to Var1 and Type 1 in metadata sheet sd
colnames(sd1) = c("iso1", "Group1")

# Join the distance matrix and the metadata sheet sd
bray_veg_intestine.melt2.sd = left_join(bray_veg_intestine.melt2, sd1, by = "iso1")

# Convert the column names to Var2 and Type 2 in metadata sheet sd
colnames(sd1) = c("iso2", "Group2")

# Join the distance matrix and the metadata sheet sd
bray_veg_intestine.melt2.sd = left_join(bray_veg_intestine.melt2.sd, sd1, by = "iso2")

write.csv(bray_veg_intestine.melt2.sd, "/Users/jaccard_Cecum.csv")

# The spreadsheet was edited to include 2 columns: group1 and then the group 2. The matrix was converted into a dataframe. 
bray_veg_intestine.melt2.sd_edited <- read.csv("/Users/jaccard_Cecum_edited.csv", 
                     header=T, 
                     row.names=1, 
                     check.names=F)
# Remove NAs
#bray.melt2.sd_lung_removed9_edited<-na.omit(bray.melt2.sd_lung_removed9_edited)

# Reorder the groups so that they are in the same order
bray_veg_intestine.melt2.sd_edited$Name<- factor(bray_veg_intestine.melt2.sd_edited$Name, levels =c('SARS2', 'FLUAV-SARS2', 'Mock', 'SARS2 v. Mock', 'FLUAV-SARS2 v. Mock','SARS2 v. FLUAV-SARS2'))

colorgroups = c("coral2", "dodgerblue1","black","darkgoldenrod3", "darkgoldenrod3", "darkgoldenrod3")

# FIGURE S7B
pdf("/Users/Boxplot_Jaccard_Cecum.pdf",
    width = 11, height = 8) 

ggplot(bray_veg_intestine.melt2.sd_edited, aes(x = Name, y = dist, color = Name)) + 
    geom_boxplot() +
    scale_color_manual(values = colorgroups)+
    scale_y_continuous(breaks = seq(0.2, 1, by=.2), limits=c(0.1, 1.1))+
    ylab("Jaccard Distance")+
    guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
    #theme_bw() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_text(vjust=1.5, size = 22),
          axis.text.x = element_text(angle=40, hjust=1, colour = "black"),
          axis.text.y = element_text(colour = "black"),
          text = element_text(size=25),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5),
          legend.position = "none")+
    theme(panel.grid = element_blank(),
            axis.line = element_line(color="black", size = 0.5),
            panel.border = element_blank(),
            panel.background =  element_rect(fill='white'))

dev.off()

stats <- compare_means(dist ~ Name,  data = bray_veg_intestine.melt2.sd_edited, method = "wilcox.test", p.adj = "bonferroni")  
stats
```

### Figure 5C, Calculate the jaccard distances and produce a boxplot for the feces
```{r}
include_list <- c("BS-540", "BS-541", "BS-542", "BS-543", "BS-545", "BS-546", "BS-547", "BS-548", "BS-549", "BS-550", "BS-551", "BS-552", "BS-553", "BS-554", "BS-556", "BS-557", "BS-558", "BS-559", "BS-560", "BS-561", "BS-562", "BS-563", "BS-564", "BS-565", "BS-566", "BS-567", "BS-568", "BS-569", "BS-570", "BS-571", "BS-572", "BS-573", "BS-574", "BS-575", "BS-576", "BS-577", "BS-578", "BS-579", "BS-580")
ASVtableveg_i_Feces <- sample.rare_i[rownames(sample.rare_i) %in% include_list, ]

# Calculating relative abundance and creating new dataframe with relative abundance data
ASVtableveg_i.rel <-         
  decostand(ASVtableveg_i_Feces, method = "total")

# Calculate the bray-curtis distances of the duodenum using vegan 
bray_veg_intestine <- vegdist(ASVtableveg_i.rel, 
                          method = "jaccard", 
                          binary = TRUE)

# Turn the distance class into a matrix  
bray_veg_intestine_matrix <- as.matrix(bray_veg_intestine)

# Use program harrietr to melt the distance matrix into 3 column data frame
bray_veg_intestine_df <- melt_dist(bray_veg_intestine_matrix)

# Convert the Var 1 and Var2 to a character 
bray_veg_intestine.melt2 = bray_veg_intestine_df %>%
  filter(as.character(iso1) != as.character(iso2)) %>%
  mutate_if(is.factor, as.character)

# Only keep columns SampleID and GroupGen
sd1 = metadata %>%
  dplyr::select("SampleID", "Group_PostInfect") %>%
  mutate_if(is.factor,as.character)

# Convert the column names to Var1 and Type 1 in metadata sheet sd
colnames(sd1) = c("iso1", "Group1")

# Join the distance matrix and the metadata sheet sd
bray_veg_intestine.melt2.sd = left_join(bray_veg_intestine.melt2, sd1, by = "iso1")

# Convert the column names to Var2 and Type 2 in metadata sheet sd
colnames(sd1) = c("iso2", "Group2")

# Join the distance matrix and the metadata sheet sd
bray_veg_intestine.melt2.sd = left_join(bray_veg_intestine.melt2.sd, sd1, by = "iso2")

write.csv(bray_veg_intestine.melt2.sd, "/Users/jaccard_Feces.csv")

# The spreadsheet was edited to include 2 columns: group1 and then the group 2. The matrix was converted into a dataframe. 
bray_veg_intestine.melt2.sd_edited <- read.csv("/Users/baseibert/Perez_Lab/Projects/Microbiome/Projects/Hamsters_SARS/R_Files/OutputFiles/jaccard_Feces_edited.csv", 
                     header=T, 
                     row.names=1, 
                     check.names=F)
# Remove NAs
#bray.melt2.sd_lung_removed9_edited<-na.omit(bray.melt2.sd_lung_removed9_edited)

# Reorder the groups so that they are in the same order
bray_veg_intestine.melt2.sd_edited$Name<- factor(bray_veg_intestine.melt2.sd_edited$Name, levels =c('SARS2', 'FLUAV-SARS2', 'Mock', 'Pre-Challenge','SARS2 v. Mock', 'SARS2 v. Pre-Challenge','FLUAV-SARS2 v. Mock', 'FLUAV-SARS2 v. Pre-Challenge','SARS2 v. FLUAV-SARS2', 'Mock v. Pre-Challenge'))

colorgroups = c("coral2", "dodgerblue1","black", "grey66","darkgoldenrod3", "darkgoldenrod3", "darkgoldenrod3", "darkgoldenrod3", "darkgoldenrod3", "darkgoldenrod3")

# Plot using ggplot with legend boxplot with points
pdf("/Users/baseibert/Boxplot_Jaccard_Feces.pdf",
    width = 16.5, height = 8) 

ggplot(bray_veg_intestine.melt2.sd_edited, aes(x = Name, y = dist, color = Name)) + 
    geom_boxplot() +
    scale_color_manual(values = colorgroups)+
    scale_y_continuous(breaks = seq(0.2, 1, by=.2), limits=c(0.1, 1.1))+
    ylab("Jaccard Distance")+
    guides(fill=guide_legend(title = "Group", override.aes=list(shape=21)))+
    #theme_bw() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_text(vjust=1.5, size = 22),
          axis.text.x = element_text(angle=40, hjust=1, colour = "black"),
          axis.text.y = element_text(colour = "black"),
          text = element_text(size=25),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5),
          legend.position = "none")+
    theme(panel.grid = element_blank(),
            axis.line = element_line(color="black", size = 0.5),
            panel.border = element_blank(),
            panel.background =  element_rect(fill='white'))

dev.off()

stats <- compare_means(dist ~ Name,  data = bray_veg_intestine.melt2.sd_edited, method = "wilcox.test", p.adj = "bonferroni")  
stats
```

# Taxonomic Analysis 

Load packages 
```{r}
library(phyloseq)
library(outliers)
library(dplyr)
library(ggplot2)
library(reshape2)
library(devtools)
library(ggpubr)
library(vegan)
library(DESeq2)
library(microbiome)
library(RColorBrewer)
library(forcats)
library(Rmisc)
library(metagMisc)
library(data.table)
```

# Figure S7C: Taxonomic bar graph of the phyla aggregated my dpi and Sample Type
```{r}
Taxonomy_i<- Phylo.samples_i_filter %>%  
  tax_glom(taxrank = "Phylum") %>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>%       # Transform to relative abundance
  psmelt()                                             # Melt to long format (this is for ggplot), create dataframe

# Multiply the abundance by 100 for a percentage
Taxonomy_i$Abundance <- Taxonomy_i$Abundance*100

Taxonomy_i_test2 <- Taxonomy_i %>%
  group_by(Group, SampleType, Phylum) %>%
  dplyr::summarise(Mean = mean(Abundance)) %>%
  arrange(-Mean)

# Reorder the groups 
Taxonomy_i_test2$Group <- factor(Taxonomy_i_test2$Group, levels = c('SARS-CoV-2', 'IAV-SARS-CoV-2','Mock'))
Taxonomy_i_test2$SampleType <- factor(Taxonomy_i_test2$SampleType, levels = c( "Duodenum", "Ileum", "Cecum", "Feces"))

colorgroups = c("lightslateblue","palevioletred1","darkgoldenrod1","paleturquoise1","green3","red2","darkseagreen","peachpuff1","mediumorchid","ivory4","tan4","royalblue2","lightsalmon2")

levels(Taxonomy_i_test2$Group)[levels(Taxonomy_i_test2$Group)=='SARS-CoV-2'] <- 'SARS2'
levels(Taxonomy_i_test2$Group)[levels(Taxonomy_i_test2$Group)=='IAV-SARS-CoV-2'] <- 'IAV-SARS2'

# FIGURE S7C
pdf("/Users/TaxonomicBarplot_phylum_AllSamples.pdf",
    width = 7, height = 9.5, # Width and height in inches
    bg = "white", )

ggplot(data=Taxonomy_i_test2, aes(x=Group, y=Mean, fill=reorder(Phylum, desc(Phylum)))) +
  geom_bar(aes(), stat = "identity", color = "black", position="stack", width = 0.9) + 
  scale_fill_manual(values=colorgroups) +
  facet_grid(~ SampleType, scales = "free", space = "fixed")+
  xlab("")+ 
  ylab("") +
  #guides(fill=guide_legend(title = "Taxonomy", ncol =1))+
  guides(fill= "none")+
  theme(axis.title.x = element_text(vjust = -.5, size = 18),
          axis.title.y = element_text(vjust=1.5, size = 18),
          axis.text.x = element_text(colour = "black", size = 13, angle=50, hjust=1),
          axis.text.y = element_text(colour = "black"),
          text = element_text(size=18),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
          #legend.title = element_text(size = 18, face = "bold"), 
          #legend.text = element_text(size = 14)) +
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.3),
          panel.border = element_blank(),
          panel.background =  element_rect(fill='white'),
         panel.spacing.x = unit(0.1, "lines"))

dev.off() 

# Plot using ggplot with legend
pdf("/Users/TaxonomicBarplot_phylum_intestine_labels.pdf",
    width = 13, height = 14, # Width and height in inches
    bg = "white", )

ggplot(data=Taxonomy_i_test2_subset, aes(x=Mean, y=Group, fill=reorder(Phylum, desc(Phylum)))) +
  geom_bar(aes(), stat = "identity", color = "black", position="stack", width = 0.9) + 
  scale_fill_manual(values=colorgroups) +
  facet_grid(dpi ~ ., scales = "free", space = "fixed")+
  xlab("Relative Abundance (%)")+ 
  ylab("") +
  guides(fill=guide_legend(title = "Taxonomy", ncol =1))+
  theme(axis.title.x = element_text(vjust = -.5, size = 24),
          axis.title.y = element_text(vjust=1.5, size = 24),
          axis.text.x = element_text(colour = "black"),
          axis.text.y = element_text(colour = "black"),
          text = element_text(size=26),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5),
          legend.title = element_text(size = 22, face = "bold"))+ 
          #legend.text = element_text(size = 14)) +
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5),
          panel.border = element_blank(),
          panel.background =  element_rect(fill='white'),
         panel.spacing.x = unit(1, "lines"))
dev.off() 
```

## Figure S7D: Plot Taxonomy barplot until the Genus Level for all tissues 
```{r}
#Import update phyloseq metadata
sample_data(Phylo.samples_i_filter) <- metadata

Taxonomy_i<- Phylo.samples_i_filter %>%  
  transform_sample_counts(function(x) {x/sum(x)} ) %>%       # Transform to relative abundance
  psmelt()                                             # Melt to long format (this is for ggplot), create dataframe

# Create a new column called TaxonomyName that combines the different taxonomic levels separated by ; 
Taxonomy_i$TaxonomyName<-paste(Taxonomy_i$Phylum,Taxonomy_i$Class, Taxonomy_i$Order, Taxonomy_i$Family, Taxonomy_i$Genus,sep="; ")
                     
# Multiply the abundance by 100 for a percentage
Taxonomy_i$Abundance <- Taxonomy_i$Abundance*100

# Filter the taxa so that those taxa that are less than 5% are grouped together 
Taxonomy_i$TaxonomyName[Taxonomy_i$Abundance < 1] <- "Taxonomy < 1 %" 
Taxonomy_i$Genus[Taxonomy_i$Abundance < 1] <- "Taxonomy < 1 %" 

# Create a new test data frame with the columns including the SampleID, Abundance, SampleType, Rank, and TaxonomyName
Taxonomy_i_test <- Taxonomy_i %>%
  group_by(Sample, SampleType, dpi, HamsterID, Group_PostInfect, TaxonomyName, Rank, Genus, Family) %>%
  dplyr::summarise(Abundance = sum(Abundance)) %>%
  as.data.frame()


# Reorder the groups 
Taxonomy_i_test$Rank <- as.factor(Taxonomy_i_test$Rank)

Taxonomy_i_test$SampleType <- factor(Taxonomy_i_test$SampleType, levels = c("Duodenum",'Ileum', 'Cecum', 'Feces'))

Taxonomy_i_test$Group_PostInfect<- factor(Taxonomy_i_test$Group_PostInfect, 
                                                          levels = c('SARS-CoV-2', 'IAV-SARS-CoV-2', 'Mock', 'Pre-Challenge'))

write.csv(Taxonomy_i_test, "/Users/TaxonomyBarplot.csv")

# Define color using R color brewer with the number of taxa we need. 

colorgroups = c("white",
                "gray85","grey60",
                "gold4","darkgoldenrod3","gold1","khaki1","lemonchiffon1",  #Proteobacteria,
                "limegreen",  #Fusobacteriota,
                "coral4",
                "indianred4","tomato4", "red4","brown3","firebrick3","red3","indianred","orangered3","red1", "indianred2", "lightcoral", 
                "violetred3", "palevioletred", "hotpink","violetred2", "palevioletred1", "hotpink","rosybrown2","pink2","pink","mistyrose",
                
                "purple4","darkmagenta","mediumorchid4","purple3", "purple", "mediumpurple1", "plum3","orchid1", "darkorchid",
                "plum1",  #Firmicutes,
                "darkseagreen", #Fibrobacter
                "burlywood2", #Elusimicrobiota
                "ivory1", #Desulfobacterota 
               "grey80", #Deferribacterota
               "darkgoldenrod4", #Camplyobacterota,
               "midnightblue","royalblue4","steelblue4","blue2", "dodgerblue3", "cornflowerblue", "deepskyblue1", "cyan1", "cadetblue1", #Bacteroidota,
               "darkorange3","orangered","chocolate1","darkorange2","lightsalmon") #Actinobacteriota

# Colors Switched
colorgroups_rev = c("lightsalmon", "darkorange2", "chocolate1", "orangered","darkorange3", #Actinobacteriota
                "cadetblue1", "cyan1", "deepskyblue1", "cornflowerblue","dodgerblue3","blue2","steelblue4","royalblue4","midnightblue",
                "darkgoldenrod4", #Camplyobacterota,
                "grey80", #Deferribacterota
                "ivory1", #Desulfobacterota 
                "burlywood2", #Elusimicrobiota
                "darkseagreen", #Fibrobacter
                "plum1", #Firmicutes,
                "darkorchid", "orchid1","plum3","mediumpurple1","purple","purple3","mediumorchid4","darkmagenta","purple4",
                "mistyrose", "pink","pink2","rosybrown2","hotpink","palevioletred1","violetred2", "hotpink","palevioletred","violetred3", 
                "lightcoral", "indianred2","red1","orangered3","indianred","red3","firebrick3","brown3","red4","tomato4", "indianred4",
                "coral4",
                "limegreen", #Fusobacteriota,
                "lemonchiffon1",  #Proteobacteria,
                "khaki1","gold1","darkgoldenrod3","gold4",
                "grey60","gray85",
                "white")

# FIGURE S7D
pdf("/Users/TaxonomicBarplot_Genus_Intestine.pdf",
    width = 13, height = 10.5, # Width and height in inches
    bg = "white", )

#fill=reorder(TaxonomyName, desc(TaxonomyName)
ggplot(data=Taxonomy_i_test, aes(x=Rank, y=Abundance, fill=TaxonomyName)) +
  geom_bar(aes(), stat = "identity", color = "black", position="stack", width = 0.9) + 
  scale_fill_manual(values=colorgroups_rev) +
  facet_grid(~SampleType, scales = "free", space = "free")+
  xlab("dpi")+ 
  ylab("Relative Abundance (%)") +
  #guides(fill=guide_legend(title = "Taxonomy", ncol =1))+
  guides(fill= "none")+
  theme(axis.title.x = element_text(vjust = -.5, size = 24),
          axis.title.y = element_text(vjust=1.5, size = 24),
          axis.text.x = element_text(colour = "black"),
          axis.text.y = element_text(colour = "black"),
          text = element_text(size=26),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
          #legend.title = element_text(size = 18, face = "bold"), 
          #legend.text = element_text(size = 14)) +
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.3),
          panel.border = element_blank(),
          panel.background =  element_rect(fill='white'))

dev.off() 
```

## Figure 5D: Plot Taxonomy heatmap until the Genus Level for all tissues 
```{r}
#Import update phyloseq metadata
sample_data(Phylo.samples_i_filter) <- metadata

Taxonomy_i<- Phylo.samples_i_filter %>%  
  transform_sample_counts(function(x) {x/sum(x)} ) %>%       # Transform to relative abundance
  psmelt()                                             # Melt to long format (this is for ggplot), create dataframe

# Create a new column called TaxonomyName that combines the different taxonomic levels separated by ; 
Taxonomy_i$TaxonomyName<-paste(Taxonomy_i$Phylum,Taxonomy_i$Class, Taxonomy_i$Order, Taxonomy_i$Family, Taxonomy_i$Genus,sep="; ")

# Change any NA to the name of the family
Taxonomy_i$Genus <- ifelse(is.na(Taxonomy_i$Genus), Taxonomy_i$Family, Taxonomy_i$Genus)

# Multiply the abundance by 100 for a percentage
Taxonomy_i$Abundance <- Taxonomy_i$Abundance*100

# Create a new test data frame with the columns including the SampleID, Abundance, SampleType, Rank, and TaxonomyName
Taxonomy_i_test <- Taxonomy_i %>%
  group_by(Sample, SampleType, dpi, HamsterID, Group_PostInfect, TaxonomyName, Rank, Genus, Family) %>%
  dplyr::summarise(Abundance = sum(Abundance)) %>%
  as.data.frame()

# Remove unnecessary genus 
keep = c('Actinomyces', 'Alloscardovia', 'Bifidobacterium','Bifidobacteriaceae', "Corynebacterium", 'Bacteroides', 'Bacteroidales RF16 group', 'Muribaculum', 'Muribaculaceae','Prevotellaceae','Prevotella', 'Prevotellaceae Ga6A1 group', 'Prevotellaceae UCG-001','Prevotellaceae UCG-003', 'Helicobacter', 'Mucispirillum', 'Desulfovibrio', 'Elusimicrobium', 'Fibrobacter', 'Anaeroplasma', 'Allobaculum', 'Ileibacterium', 'Erysipelotrichaceae', 'Granulicatella', 'Carnobacteriaceae', 'Ligilactobacillus', 'Streptococcus', 'Mycoplasma', 'Staphylococcus', 'Eubacteriaceae', 'Acetatifactor', 'Anaerostipes','Lachnospiraceae NK4A136 group','Lachnospiraceae', 'Tyzzerella', '[Eubacterium] coprostanoligenes group', 'NK4A214 group', 'Oscillibacter','UCG-003','UCG-005','[Eubacterium] siraeum group', 'Ruminococcaceae', 'Ruminococcus', 'Peptococcus', 'S5-A14a', 'Finegoldia', 'Peptoniphilus', 'Peptostreptococcus', 'Quinella', 'Megasphaera', 'Veillonella', 'Fusobacterium','Parasutterella','Haemophilus','Mannheimia', 'Pasteurellaceae', 'Rodentibacter', 'Spirochaetaceae', 'Treponema')
    
Taxonomy_i_edited <- filter(Taxonomy_i_test,
                       Genus %in% keep)

# Change any NA to the name of the family
Taxonomy_i_edited$Genus <- ifelse(is.na(Taxonomy_i_edited$Genus), Taxonomy_i_edited$Family, Taxonomy_i_edited$Genus)

# transform the table so that you add the aggregated abundances 
Agg_intestine <- aggregate(Abundance ~ Genus + Sample + Rank + Group_PostInfect + dpi + SampleType + TaxonomyName, data = Taxonomy_i_edited, sum)

Agg_intestine$Genus <- factor(Agg_intestine$Genus, levels = c('Actinomyces', 'Alloscardovia', 'Bifidobacterium',"Corynebacterium", 'Bifidobacteriaceae',
              
'Bacteroides', 'Muribaculum', 'Prevotella', 'Prevotellaceae Ga6A1 group', 'Prevotellaceae UCG-001','Prevotellaceae UCG-003', 'Bacteroidales RF16 group', 'Muribaculaceae', 'Prevotellaceae',


'Helicobacter', 

'Mucispirillum', 

'Desulfovibrio', 

'Elusimicrobium', 

'Fibrobacter', 

'Allobaculum', 'Anaeroplasma', 'Granulicatella', 'Ileibacterium', 'Ligilactobacillus', 'Mycoplasma', 'Streptococcus', 'Staphylococcus', 'Carnobacteriaceae', 'Erysipelotrichaceae', 
'Acetatifactor', 'Anaerostipes','[Eubacterium] siraeum group',  'Finegoldia', 'Lachnospiraceae NK4A136 group', 'NK4A214 group', 'Oscillibacter', 'Peptococcus','Peptoniphilus', 'Peptostreptococcus', 'Ruminococcus', 'S5-A14a', 'Tyzzerella', 'UCG-003','UCG-005', 'Eubacteriaceae', 'Lachnospiraceae','[Eubacterium] coprostanoligenes group',  'Ruminococcaceae', 
'Quinella', 'Megasphaera', 'Veillonella', 

'Fusobacterium',

'Haemophilus','Mannheimia', 'Parasutterella', 'Rodentibacter', 'Pasteurellaceae',

'Treponema', 'Spirochaetaceae'))

Agg_intestine$Rank <- as.factor(Agg_intestine$Rank)

Agg_intestine$Group_PostInfect <- factor(Agg_intestine$Group_PostInfect, levels = c('SARS-CoV-2', 'IAV-SARS-CoV-2', 'Mock', 'Pre-Challenge'))

Agg_intestine$SampleType <- factor(Agg_intestine$SampleType, levels = c('Duodenum','Ileum', 'Cecum', 'Feces'))

theme_set(theme_bw(20))

# FIGURE 5D
# Plot using ggplot with legend
pdf("/Users/RA_Intestine_Heatmap_AllSamples.pdf",
    width = 18, height =6.5, 
    bg = NA) # Width and height in inches

ggplot(Agg_intestine, aes(Rank, fct_rev(Genus), group = Group_PostInfect)) +
    geom_tile(aes(fill = Abundance), color = "darkblue", lwd = 0.1) +
    scale_fill_gradientn("Abundance", 
                      breaks = seq(from = 0, to = 10,  by = 2), 
                      colours = c("blue", "yellow", "red"), 
                  limits = c(0, 10), 
                  na.value = "green") +
  facet_grid(~SampleType, scales = "free", space = "free")+
    theme(axis.title.x=element_blank(),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black", size = 9),
          axis.text.y = element_text(colour = "black", size = 10, face = "italic"),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5),
          legend.title = element_blank(), 
          legend.text = element_text(size = 8)) +
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5), 
          legend.position="right", 
          legend.key.width = unit(0.6, 'cm'))

dev.off()
```

# Figure S8 (middle): Differential Analysis in the intestine and feces between IAV-SARS-CoV-2 and Mock groups using ALDEx2 at the Genus level 
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown (This can be used to produce the different graphs in Fig S8)
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  subset_samples(SampleType == "Duodenum" | SampleType == "Ileum") %>%
#  subset_samples(SampleType == "Cecum") %>%
#  subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Pre-Challenge")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited, Genus~SampleID, value.var = "Abundance", sum)

#Subset the pre-challenge groups to only include those hamsters that were challenged 
#IAVSARS challenged
Phylo.samples_i_filter_10_sub_df_edited2 <- Phylo.samples_i_filter_10_sub_df_edited %>%
  filter(HamsterID == "11" | HamsterID == "12" | HamsterID == "13" |HamsterID == "14" |HamsterID == "15" | HamsterID == "16" | HamsterID == "17")

#SARS challenged
Phylo.samples_i_filter_10_sub_df_edited2 <- Phylo.samples_i_filter_10_sub_df_edited %>%
  filter(HamsterID == "4" | HamsterID == "5" | HamsterID == "7" |HamsterID == "8" |HamsterID == "9" | HamsterID == "10")

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited2, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
# add 1 to all values 
dc_matrix_plus1 = dc + 1

# Define the conditions 
#conds <- c(rep("Mock", 3), rep("IAVSARS", 7), rep("Mock", 3), rep("IAVSARS", 7),rep("Mock", 3), rep("IAVSARS", 7),rep("Mock", 3), rep("IAVSARS", 7), rep("Mock", 2), rep("IAVSARS", 4)) #All Samples Mock v IAV

#conds <- c(rep("Mock", 3), rep("IAVSARS", 7), rep("Mock", 3), rep("IAVSARS", 7)) #SmallIntestine Mock v IAV

#conds <- c(rep("Mock", 3), rep("IAVSARS", 7)) #Cecum Mock v IAV

#conds <- c(rep("Mock", 3), rep("IAVSARS", 7), rep("Mock", 2), rep("IAVSARS", 4))  #Feces Mock v IAV

#conds <- c(rep("Mock", 3), rep("SARS", 6), rep("Mock", 3), rep("SARS", 6), rep("Mock", 3), rep("SARS", 6), rep("Mock", 3), rep("SARS", 6), rep("Mock", 2), rep("SARS", 3)) #All Samples Mock v SARS

#conds <- c(rep("Mock", 3), rep("SARS", 6), rep("Mock", 3), rep("SARS", 6)) #SmallIntestine Mock v SARS

#conds <- c(rep("Mock", 3), rep("SARS", 6)) #Cecum Mock v SARS

#conds <- c(rep("Mock", 3), rep("SARS", 6), rep("Mock", 2), rep("SARS", 3))  #Feces Mock v SARS

#conds <- c(rep("SARS", 6), rep("IAVSARS", 7), rep("SARS", 6), rep("IAVSARS", 7),rep("SARS", 6), rep("IAVSARS", 7), rep("SARS", 6), rep("IAVSARS", 7), rep("SARS", 3), rep("IAVSARS", 4)) #All Samples Mock v IAV

#conds <- c(rep("SARS", 6), rep("IAVSARS", 7), rep("SARS", 6), rep("IAVSARS", 7)) #SmallIntestine SARS v IAVSARS

#conds <- c(rep("SARS", 6), rep("IAVSARS", 7)) #Cecum SARS v IAVSARS

#conds <- c(rep("SARS", 6), rep("IAVSARS", 7), rep("SARS", 3), rep("IAVSARS", 4)) #Feces SARS v IAVSARS

#conds <- c(rep("Pre-Challenge", 16), rep("Mock", 5)) #Feces Mock v Prechallenge

#conds <- c(rep("PreChallenge", 7), rep("IAVSARS", 11)) #Feces Prechallenge v IAVSARS

#conds <- c(rep("PreChallenge", 6), rep("SARS", 9)) #Feces Prechallenge v SARS

conds <- c(rep("Control", 9), rep("IAV", 7)) #Feces Prechallenge controls versus those that were exposed to IAV previously

# Perform aldex on pairwise comparison. Most basic function 
#aldexMock_IAVSARS <- aldex(dc, conds, mc.samples = 128, test="t", effect=TRUE, verbose = FALSE)

# The workflow for the modular approach first generates random instances of the centred log-ratio transformed values. There are three inputs: counts table, a vector of conditions, and the number of Monte-Carlo (DMC) instances; and several parameters: a string indicating if iqlr, zero or all feature are used as the denominator is required, and level of verbosity (TRUE or FALSE). We recommend 128 or more mc.samples for the t-test, 1000 for a rigorous effect size calculation, and at least 16 for ANOVA
aldexMock_IAVSARS_clr <- aldex.clr(dc_matrix_plus1, conds, mc.samples = 128, verbose = TRUE)

# The next operation performs the Welchâ€™s t and Wilcoxon rank test for the situation when there are only two conditions. There are only two inputs: the aldex object from aldex.clr and whether a paired test should be conducted or not (TRUE or FALSE).
x.tt <- aldex.ttest(aldexMock_IAVSARS_clr, paired.test=FALSE, hist.plot = TRUE, verbose=TRUE)

# Finally, we estimate effect size and the within and between condition values in the case of two conditions. This step is required for plotting, in our lab we base our conclusions primarily on the output of this function.There is one input: the aldex object from aldex.clr, ; and several parameters: a flag as to whether to include values for all samples or not are used as the denominator, and the level of verbosity. It is also possible to include the 95% confidence interval information for the effect size estimate with the flag CI=TRUE. This can be helpful when deciding whether to include or exclude specific features from consideration. We find that a large effect but that is an outlier for the extremes of the effect distribution can be false positives.
x.effect <- aldex.effect(aldexMock_IAVSARS_clr, CI=T, verbose=TRUE)
summary(x.effect)
# the t-test and effect data are merged into one object.
x.all_Mock_IAVSARS <- data.frame(x.tt,x.effect)

write.csv(x.all_Mock_IAVSARS,"/Users/ALDeX2_Genus_AllSamples_PreChallenge_ControlvsIAV_Output.csv")

pdf("/Users/ALDeX2_Genus_AllSamples_PreChallenge_ControlvIAV_MW.pdf",
    width = 3.5, height = 4, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

#par(mfrow=c(1,2))
#aldex.plot(x.all_Mock_IAVSARS, type="MA", test="welch", xlab="Log-ratio abundance",
#    ylab="Difference")
aldex.plot(x.all_Mock_IAVSARS, type="MW", test="wilcox", cutoff = 2, called.col = "red")

dev.off()

# identify which values are significant in wilcox
signocorrect <- which(x.all_Mock_IAVSARS$wi.ep < 0.05)
signocorrect

# identify which values are significant in wilcox
sig <- which(x.all_Mock_IAVSARS$wi.eBH < 0.05)
sig
# identify which values have an effect size of greater than 2 and have a effect size CI that does not overlap 0 
effectsize <- which(x.all_Mock_IAVSARS$effect > 1.5 | x.all_Mock_IAVSARS$effect.low > 0)
effectsize

pdf("/Users/ALDeX2_Genus_AllSamples_Prechallenge_ControlvIAV_MW_Edited.pdf",
    width = 3.5, height = 4, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

plot(x.all_Mock_IAVSARS$diff.win, x.all_Mock_IAVSARS$diff.btw, pch=19, cex=0.4, col = "gray54", 
     xlab="Difference within", ylab="Difference between")
points(x.all_Mock_IAVSARS$diff.win[signocorrect], x.all_Mock_IAVSARS$diff.btw[signocorrect], pch=21, cex=0.4, bg="limegreen", col = "limegreen")
points(x.all_Mock_IAVSARS$diff.win[sig], x.all_Mock_IAVSARS$diff.btw[sig], pch=21, cex=0.4, bg="red", col = "red")
points(x.all_Mock_IAVSARS$diff.win[effectsize], x.all_Mock_IAVSARS$diff.btw[effectsize], pch=21, cex=0.5, col = "blue")
abline(0,1,lty=2)
abline(0,-1,lty=2)

dev.off()

```

# Figure S8A (left): Differential Analysis in the intestine/feces between Mock and the SARS2 groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  #subset_samples(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  #subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Mock" | Group_PostInfect == "SARS-CoV-2")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")

metadata <- read.csv("/Users/Sars_hamster_metadata.csv")
# remove unnecessary columns
metadata_Intestine <- metadata %>%
  filter(SampleType == "Duodenum" | SampleType == "Ileum" | SampleType == "Cecum" | SampleType == "Feces")

metadata_Intestine <- metadata %>%
  filter(SampleType == "Duodenum" | SampleType == "Ileum")

metadata_Intestine <- metadata %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "Mock" | Group_PostInfect == "SARS-CoV-2") 

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,7)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ DeseqGroup)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$DeseqGroup, "SARSCoV2" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("DeseqGroup_Mock_vs_IAV_SARSCoV2")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/Deseq2_Analysis_Feces_IAVSARS.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)

sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Bifidobacteriaceae","Unclassified Bifidobacteriaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Prevotellaceae","Unclassified Prevotellaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Lachnospiraceae","Unclassified Lachnospiraceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Spirochaetaceae","Unclassified Spirochaetaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Christensenellaceae","Unclassified Christensenellaceae")

# Order the Genus 
sigtab05_table$Genus <- factor(sigtab05_table$Genus, levels = c("Unclassified Bifidobacteriaceae", #Actinobacteria
  "Alistipes", "Parabacteroides", "Unclassified Prevotellaceae", #Bacteroidetes
  "Elusimicrobium", #Elusimicrobiota
  "Acetatifactor","Allobaculum", "Christensenellaceae R-7 group","[Eubacterium] siraeum group", "Ligilactobacillus", 
  "NK4A214 group", "Streptococcus","UCG-005", "Veillonella", "Unclassified Christensenellaceae", "Unclassified Lachnospiraceae", #Firmicutes
  "Unclassified Spirochaetaceae")) 


# Graph the log2foldchange 
bxp <- ggplot(sigtab05_table, aes(x = log2FoldChange, y = fct_rev(Genus), fill = log2FoldChange > 0))+
  geom_col(colour = "Black")+
  scale_fill_manual(name = 'log2FoldChange > 0', values = setNames(c('coral2','grey66'),c(T, F)))+
  theme_minimal()+
  geom_vline(xintercept=c(-10,-5,0,5,10), color = "gray63", size = 0.2)+
  theme(axis.title.x = element_text(colour = "black", size = 11),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black",size = 10),
          axis.text.y = element_text(colour = "black", size = 10),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
    theme(axis.line = element_line(color="black", size = 0.5),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(), 
        legend.position = "none")
bxp 

dp <- ggplot(sigtab05_table, aes(x = lfcSE, y = fct_rev(Genus)))+
  geom_point() +
  scale_x_continuous(breaks = seq(0, 2.5, by=0.5), limits=c(0, 2.5))+
  theme_minimal()+
  theme(axis.text.y=element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(colour = "black"), 
        axis.title.x = element_text(colour = "black"))+
  theme(axis.line = element_line(color="black", size = 0.4))
dp  

# FIGURE 8 LEFT
pdf("/Users/Deseq2_Genus_AllSamples_MockSARS.pdf",
    width = 10.5, height =4.7, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

ggarrange(bxp, dp,
  widths = c(3,1))

dev.off()
```

# Figure S8B (left): Differential Analysis in the intestine/feces between Mock and the FLUAV-SARS2 groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  #subset_samples(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  #subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Mock" | Group_PostInfect == "SARS-CoV-2")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")
# remove unnecessary columns
metadata_Intestine <- metadata %>%
  filter(SampleType == "Duodenum" | SampleType == "Ileum" | SampleType == "Cecum" | SampleType == "Feces")

metadata_Intestine <- metadata %>%
  filter(SampleType == "Duodenum" | SampleType == "Ileum")

metadata_Intestine <- metadata %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "Mock" | Group_PostInfect == "SARS-CoV-2") 

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,7)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ DeseqGroup)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$DeseqGroup, "SARSCoV2" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("DeseqGroup_Mock_vs_SARSCoV2")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/TaxonomicAnalysis/Deseq2_Analysis_Feces_MockSARS.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)

sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Bifidobacteriaceae","Unclassified Bifidobacteriaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Prevotellaceae","Unclassified Prevotellaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Lachnospiraceae","Unclassified Lachnospiraceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Spirochaetaceae","Unclassified Spirochaetaceae")

# Order the Genus 
sigtab05_table$Genus <- factor(sigtab05_table$Genus, levels = c(
  "Alistipes", "Prevotellaceae UCG-001", "Unclassified Prevotellaceae", #Bacteroidetes
  "Elusimicrobium", #Elusimicrobiota
  "Fibrobacter", #Fibrobacter
  "Anaerostipes", "Ligilactobacillus","Mycoplasma","NK4A214 group", "Streptococcus", "Veillonella", "Unclassified Lachnospiraceae", #Firmicutes
  "Unclassified Spirochaetaceae")) 


# Graph the log2foldchange 
bxp <- ggplot(sigtab05_table, aes(x = log2FoldChange, y = fct_rev(Genus), fill = log2FoldChange > 0))+
  geom_col(colour = "Black")+
  scale_fill_manual(name = 'log2FoldChange > 0', values = setNames(c('coral2','grey66'),c(T, F)))+
  theme_minimal()+
  geom_vline(xintercept=c(-10,-5,0,5,10), color = "gray63", size = 0.2)+
  theme(axis.title.x = element_text(colour = "black", size = 11),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black",size = 10),
          axis.text.y = element_text(colour = "black", size = 10),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
    theme(axis.line = element_line(color="black", size = 0.5),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(), 
        legend.position = "none")
bxp 

dp <- ggplot(sigtab05_table, aes(x = lfcSE, y = fct_rev(Genus)))+
  geom_point() +
  scale_x_continuous(breaks = seq(0, 2.5, by=0.5), limits=c(0, 2.5))+
  theme_minimal()+
  theme(axis.text.y=element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(colour = "black"), 
        axis.title.x = element_text(colour = "black"))+
  theme(axis.line = element_line(color="black", size = 0.4))
dp 

pdf("/Users/Deseq2_Genus_AllSamples_MockSARS.pdf",
    width = 11, height =5, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

ggarrange(bxp, dp,
  widths = c(3,1))

dev.off()
```

# Figure S8C (left): Differential Analysis in the intestine/feces between FLUAV-SARS2 and the SARS2 groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  #subset_samples(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  #subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "SARS-CoV-2" | Group_PostInfect == "IAV-SARS-CoV-2")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")
# remove unnecessary columns
metadata_Intestine <- metadata %>%
  filter(SampleType == "Duodenum" | SampleType == "Ileum" | SampleType == "Cecum" | SampleType == "Feces")

metadata_Intestine <- metadata %>%
  filter(SampleType == "Duodenum" | SampleType == "Ileum")

metadata_Intestine <- metadata %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "SARS-CoV-2" | Group_PostInfect == "IAV-SARS-CoV-2") 

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,7)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ DeseqGroup)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$DeseqGroup, "SARSCoV2" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("DeseqGroup_SARSCoV2_vs_IAV_SARSCoV2")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/Deseq2_Analysis_Feces_SARSIAV.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)

sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Bifidobacteriaceae","Unclassified Bifidobacteriaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Ruminococcaceae","Unclassified Ruminococcaceae")

# Order the Genus 
sigtab05_table$Genus <- factor(sigtab05_table$Genus, levels = c("Unclassified Bifidobacteriaceae", #Actinobacter
  "Fibrobacter", #Fibrobacter
  "Anaerostipes", "Lachnoclostridium", "UCG-005", "Unclassified Ruminococcaceae", #Firmicutes
  "Coxiella", "Haemophilus" #Proteobacteria
  )) 


# Graph the log2foldchange 
bxp <- ggplot(sigtab05_table, aes(x = log2FoldChange, y = fct_rev(Genus), fill = log2FoldChange > 0))+
  geom_col(colour = "Black")+
  scale_fill_manual(name = 'log2FoldChange > 0', values = setNames(c('dodgerblue1','coral2'),c(T, F)))+
  theme_minimal()+
  geom_vline(xintercept=c(-10, -5,-0, 5, 10), color = "gray63", size = 0.2)+
  theme(axis.title.x = element_text(colour = "black", size = 11),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black",size = 10),
          axis.text.y = element_text(colour = "black", size = 10),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
    theme(axis.line = element_line(color="black", size = 0.5),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(), 
        legend.position = "none")
bxp 

dp <- ggplot(sigtab05_table, aes(x = lfcSE, y = fct_rev(Genus)))+
  geom_point() +
  scale_x_continuous(breaks = seq(0, 2.5, by=0.5), limits=c(0, 2.5))+
  theme_minimal()+
  theme(axis.text.y=element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(colour = "black"), 
        axis.title.x = element_text(colour = "black"))+
  theme(axis.line = element_line(color="black", size = 0.4))
dp 

pdf("/Users/Deseq2_Genus_AllSamples_SARSIAV.pdf",
    width = 11, height =5, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

ggarrange(bxp, dp,
  widths = c(3,1))

dev.off()
```

# Differential Analysis in the feces between Pre-Challenge and Mock groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  #subset_samples(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Pre-Challenge" | Group_PostInfect == "Mock")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")
# remove unnecessary columns
metadata3 <- read.csv("/Users/Sars_hamster_metadata_MockPreChallenge.csv")

metadata_Intestine <- metadata3 %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "Pre-Challenge" | Group_PostInfect == "Mock") 

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,7)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ DeseqGroup)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$DeseqGroup, "Mock" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("DeseqGroup_SARSCoV2_vs_IAV_SARSCoV2")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/TaxonomicAnalysis/Deseq2_Analysis_Feces_MockPrechallenge.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)

sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Spirochaetaceae","Unclassified Spirochaetaceae")

# Order the Genus 
sigtab05_table$Genus <- factor(sigtab05_table$Genus, levels = c(
  "Alistipes", #Bacter
  "Elusimicrobium",
  "Ligilactobacillus", "NK4A214 group", #Firmicutes
  "Mannheimia", 
  "Unclassified Spirochaetaceae" #Proteobacteria
  )) 


# Graph the log2foldchange 
bxp <- ggplot(sigtab05_table, aes(x = log2FoldChange, y = fct_rev(Genus), fill = log2FoldChange > 0))+
  geom_col(colour = "Black")+
  scale_fill_manual(name = 'log2FoldChange > 0', values = setNames(c('grey20','grey66'),c(T, F)))+
  theme_minimal()+
  geom_vline(xintercept=c(-10,-5,0,5,10), color = "gray63", size = 0.2)+
  theme(axis.title.x = element_text(colour = "black", size = 11),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black",size = 10),
          axis.text.y = element_text(colour = "black", size = 10),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
    theme(axis.line = element_line(color="black", size = 0.5),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(), 
        legend.position = "none")
bxp 

dp <- ggplot(sigtab05_table, aes(x = lfcSE, y = fct_rev(Genus)))+
  geom_point() +
  scale_x_continuous(breaks = seq(0, 2.5, by=0.5), limits=c(0, 2.5))+
  theme_minimal()+
  theme(axis.text.y=element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(colour = "black"), 
        axis.title.x = element_text(colour = "black"))+
  theme(axis.line = element_line(color="black", size = 0.4))
dp 

pdf("/Users/TaxonomicAnalysis/Deseq2_Genus_AllSamples_MockPreChallenge.pdf",
    width = 11, height =5, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

ggarrange(bxp, dp,
  widths = c(3,1))

dev.off()
```

# Differential Analysis in the feces between Pre-Challenge and Mock groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  #subset_samples(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Pre-Challenge")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

#Subset the pre-challenge groups to only include those hamsters that were challenged 

# Classify mock and pre-challenge
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df %>% mutate(GroupPreChallenge =
                     case_when(HamsterID == "1" ~ "Mock", 
                               HamsterID == "2" ~ "Mock",
                               HamsterID == "3" ~ "Mock",
                               HamsterID == "4" ~ "Mock",
                               HamsterID == "5" ~ "Mock",
                               HamsterID == "7" ~ "Mock",
                               HamsterID == "8" ~ "Mock",
                               HamsterID == "9" ~ "Mock",
                               HamsterID == "10" ~ "Mock",
                               HamsterID == "11" ~ "FLUAV",
                               HamsterID == "12" ~ "FLUAV",
                               HamsterID == "13" ~ "FLUAV",
                               HamsterID == "14" ~ "FLUAV",
                               HamsterID == "15" ~ "FLUAV",
                               HamsterID == "16" ~ "FLUAV",
                               HamsterID == "17" ~ "FLUAV")
)

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited2 <- Phylo.samples_i_filter_10_sub_df_edited
Phylo.samples_i_filter_10_sub_df_edited2$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited2$Genus), Phylo.samples_i_filter_10_sub_df_edited2$Family, Phylo.samples_i_filter_10_sub_df_edited2$Genus)

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited2, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")
# remove unnecessary columns
metadata3 <- read.csv("/Users/Sars_hamster_metadata_MockPreChallenge.csv")

metadata_Intestine <- metadata3 %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "Pre-Challenge") 

# add a new column classifying all animals that did not get FLUAV and then those that were pre-exposed to FLUAV
metadata_Intestine <- metadata_Intestine %>% mutate(GroupPreChallenge =
                     case_when(Group == "Mock" ~ "Mock", 
                               Group == "SARS-CoV-2" ~ "Mock", 
                               Group == "IAV-SARS-CoV-2" ~ "FLUAV")
)

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,18)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ GroupPreChallenge)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$GroupPreChallenge, "Mock" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("GroupPreChallenge_Mock_vs_FLUAV")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/Deseq2_Analysis_Feces_MockPrechallengevFLUAVPreChallenge.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)

# Order the Genus 
sigtab05_table$Genus <- factor(sigtab05_table$Genus, levels = c(
  "Anaerostipes", #Bacter
  "[Ruminococcus] torques group"
  )) 


# Graph the log2foldchange 
bxp <- ggplot(sigtab05_table, aes(x = log2FoldChange, y = fct_rev(Genus), fill = log2FoldChange > 0))+
  geom_col(colour = "Black")+
  scale_fill_manual(name = 'log2FoldChange > 0', values = setNames(c('grey20','grey66'),c(T, F)))+
  theme_minimal()+
  geom_vline(xintercept=c(-10,-5,0,5,10), color = "gray63", size = 0.2)+
  theme(axis.title.x = element_text(colour = "black", size = 11),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black",size = 10),
          axis.text.y = element_text(colour = "black", size = 10),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
    theme(axis.line = element_line(color="black", size = 0.5),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(), 
        legend.position = "none")
bxp 

dp <- ggplot(sigtab05_table, aes(x = lfcSE, y = fct_rev(Genus)))+
  geom_point() +
  scale_x_continuous(breaks = seq(0, 2.5, by=0.5), limits=c(0, 2.5))+
  theme_minimal()+
  theme(axis.text.y=element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(colour = "black"), 
        axis.title.x = element_text(colour = "black"))+
  theme(axis.line = element_line(color="black", size = 0.4))
dp 

pdf("/Users//TaxonomicAnalysis/Deseq2_Genus_AllSamples_MockPreChallenge.pdf",
    width = 11, height =5, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

ggarrange(bxp, dp,
  widths = c(3,1))

dev.off()
```

# Differential Analysis in the feces between PreChallenge and FLUAV-SARS-CoV-2 groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Pre-Challenge" | Group_PostInfect == "IAV-SARS-CoV-2")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

#Subset the pre-challenge groups to only include those hamsters that were challenged 
#IAVSARS challenged
Phylo.samples_i_filter_10_sub_df_edited2 <- Phylo.samples_i_filter_10_sub_df_edited %>%
  filter(HamsterID == "11" | HamsterID == "12" | HamsterID == "13" |HamsterID == "14" |HamsterID == "15" | HamsterID == "16" | HamsterID == "17")

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited2, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")

metadata <- read.csv("/Users/Sars_hamster_metadata.csv")
# remove unnecessary columns
metadata_Intestine <- metadata %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "Pre-Challenge" | Group_PostInfect == "IAV-SARS-CoV-2") 

metadata_Intestine <- metadata_Intestine %>%
  filter(HamsterID == "11" | HamsterID == "12" | HamsterID == "13" |HamsterID == "14" |HamsterID == "15" | HamsterID == "16" | HamsterID == "17")

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,7)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ DeseqGroup)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$DeseqGroup, "IAV_SARSCoV2" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("DeseqGroup_PreChallenge_vs_IAV_SARSCoV2")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/Deseq2_Analysis_Feces_PreChallengeIAVSARS.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)
```

# Differential Analysis in the feces between PreChallenge and SARS-CoV-2 groups Using Deseq2
```{r}
# Filter OTUs that are found in at least 10% of samples and total OTU abundance is >= 10 reads 
library(metagMisc)
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

# Filter the data so that only 2 groups are shown 
Phylo.samples_i_filter_10_sub <- Phylo.samples_i_filter_10 %>%
  subset_samples(SampleType == "Feces") %>%
  subset_samples(Group_PostInfect == "Pre-Challenge" | Group_PostInfect == "SARS-CoV-2")
Phylo.samples_i_filter_10_sub

Phylo.samples_i_filter_10_sub_df <- Phylo.samples_i_filter_10_sub %>%  
  psmelt() 

# Change any NA to the name of the family
Phylo.samples_i_filter_10_sub_df_edited <- Phylo.samples_i_filter_10_sub_df
Phylo.samples_i_filter_10_sub_df_edited$Genus <- ifelse(is.na(Phylo.samples_i_filter_10_sub_df_edited$Genus), Phylo.samples_i_filter_10_sub_df_edited$Family, Phylo.samples_i_filter_10_sub_df_edited$Genus)

#Subset the pre-challenge groups to only include those hamsters that were challenged 
#SARS challenged
Phylo.samples_i_filter_10_sub_df_edited2 <- Phylo.samples_i_filter_10_sub_df_edited %>%
  filter(HamsterID == "4" | HamsterID == "5" | HamsterID == "7" |HamsterID == "8" |HamsterID == "9" | HamsterID == "10")

# transform the table so that you add the aggregated abundances 
dc <- dcast(Phylo.samples_i_filter_10_sub_df_edited2, Genus~SampleID, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc) <- dc[,1]
# remove first column 
dc <- dc[,-1]
dc_matrix <- data.matrix(dc, rownames.force = TRUE)
# add 1 to all values 
dc_matrix_plus1 = dc_matrix + 1

# Make the second file (first column being sample names, and the other columns being factors, like "treatment" or "time point" or "date of sequencing" or "species")

metadata <- read.csv("/Users/Sars_hamster_metadata.csv")
# remove unnecessary columns
metadata_Intestine <- metadata %>%
  filter(SampleType == "Feces")

metadata_Intestine <- metadata_Intestine %>%
  filter(Group_PostInfect == "Pre-Challenge" | Group_PostInfect == "SARS-CoV-2") 

metadata_Intestine <- metadata_Intestine %>%
  filter(HamsterID == "4" | HamsterID == "5" | HamsterID == "7" |HamsterID == "8" |HamsterID == "9" | HamsterID == "10")

metadata_Intestine_metadata_edit <- metadata_Intestine[,c(2,7)]
metadata_Intestine_metadata_edit <- metadata_Intestine_metadata_edit %>%
  arrange(SampleID)

# Construct Deseq table 
ddsFullCountTable_plus1 <- DESeqDataSetFromMatrix(
    countData = dc_matrix_plus1,
    colData = metadata_Intestine_metadata_edit,
    design = ~ DeseqGroup)
ddsFullCountTable_plus1

# Relevel Deseq table so that Mock is considered the control
ddsFullCountTable_plus1$DeseqGroup <- relevel(ddsFullCountTable_plus1$DeseqGroup, "PreChallenge" )

dds <- DESeq(ddsFullCountTable_plus1, fitType = "local") 
plotDispEsts(dds)

# check levels of the Deseq ("DeseqGroup_SARSCoV2_vs_PreChallenge")
resultsNames(dds)

# Create a Results table 
res <- results(dds, cooksCutoff = FALSE) 
res
res = res[order(res$padj, na.last=NA), ]
sigtab05 = res[(res$padj < 0.01), ]
dim(sigtab05)

sigtab05_table = cbind(as(sigtab05, "data.frame"))

res_table = cbind(as(res, "data.frame"))

#Write out as csv
write.csv(res_table,"/Users/Deseq2_Analysis_Feces_PreChallengeSARS.csv")

# Add a column showing the genus 
sigtab05_table$Genus <- row.names(sigtab05_table)

sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Bifidobacteriaceae","Unclassified Bifidobacteriaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Prevotellaceae","Unclassified Prevotellaceae")
sigtab05_table$Genus <-replace(sigtab05_table$Genus,sigtab05_table$Genus == "Eubacteriaceae","Unclassified Eubacteriaceae")

# Order the Genus 
sigtab05_table$Genus <- factor(sigtab05_table$Genus, levels = c("Bifidobacterium","Unclassified Bifidobacteriaceae", #Actinobacteria
  "Prevotella", "Prevotellaceae UCG-001", "Unclassified Prevotellaceae", #Bacteroidetes
  "Fibrobacter", 
  "Allobaculum","Faecalibaculum","Ileibacterium",
  "Lachnospiraceae NK4A136 group","Unclassified Eubacteriaceae")) #Firmicutes

# Graph the log2foldchange 
bxp <- ggplot(sigtab05_table, aes(x = log2FoldChange, y = fct_rev(Genus), fill = log2FoldChange > 0))+
  geom_col(colour = "Black")+
  scale_fill_manual(name = 'log2FoldChange > 0', values = setNames(c('dodgerblue1','grey66'),c(T, F)))+
  theme_minimal()+
  geom_vline(xintercept=c(-10,-5,0,5,10), color = "gray63", size = 0.2)+
  theme(axis.title.x = element_text(colour = "black", size = 11),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black",size = 10),
          axis.text.y = element_text(colour = "black", size = 10),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5))+
    theme(axis.line = element_line(color="black", size = 0.5),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(), 
        legend.position = "none")
bxp 

dp <- ggplot(sigtab05_table, aes(x = lfcSE, y = fct_rev(Genus)))+
  geom_point() +
  scale_x_continuous(breaks = seq(0, 2.5, by=0.5), limits=c(0, 2.5))+
  theme_minimal()+
  theme(axis.text.y=element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(colour = "black"), 
        axis.title.x = element_text(colour = "black"))+
  theme(axis.line = element_line(color="black", size = 0.4))
dp  

pdf("/Users/Deseq2_Genus_AllSamples_PreChallengeSARS.pdf",
    width = 10.5, height =4.7, # Width and height in inches (3.5 x 4 for single graph) (7 x 4 for double graph)
    bg = "white", )

ggarrange(bxp, dp,
  widths = c(3,1))

dev.off()
```

# Figure S8 (right): Produce the file to submit for Lefse analysis
```{r}
#Phylo.samples_l_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_l_filter, prev.trh = 0.2)
Phylo.rare_i_filter_10 <- phyloseq_filter_prevalence(Phylo.rare_i, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.rare_i
Phylo.rare_i_filter_10

Taxonomy_i_lefse <- Phylo.rare_i_filter_10 %>%       # Transform to relative abundance
  psmelt()                                             # Melt to long format (this is for ggplot), create dataframe

Taxonomy_i_lefse[is.na(Taxonomy_i_lefse)] <- "" 

# Create a new column called TaxonomyName that combines the different taxonomic levels separated by ; 
Taxonomy_i_lefse$TaxonomyName<-paste(Taxonomy_i_lefse$Kingdom, Taxonomy_i_lefse$Phylum,Taxonomy_i_lefse$Class, Taxonomy_i_lefse$Order, Taxonomy_i_lefse$Family, Taxonomy_i_lefse$Genus,sep="|")

Phylo.samples_i_rare_10_lefse <- Taxonomy_i_lefse %>%
  #filter(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  filter(SampleType == "Feces") %>%
  filter(Group_PostInfect == "Pre-Challenge")

#Subset the pre-challenge groups to only include those hamsters that were challenged 
#IAVSARS challenged
Phylo.samples_i_rare_10_lefse <- Phylo.samples_i_rare_10_lefse %>%
  filter(HamsterID == "11" | HamsterID == "12" | HamsterID == "13" |HamsterID == "14" |HamsterID == "15" | HamsterID == "16" | HamsterID == "17")

#SARS challenged
Phylo.samples_i_rare_10_lefse <- Phylo.samples_i_rare_10_lefse %>%
  filter(HamsterID == "4" | HamsterID == "5" | HamsterID == "7" |HamsterID == "8" |HamsterID == "9" | HamsterID == "10")

# Rename all NA to unclassified 
#Taxonomy_l_removed9_taxglom_edit[is.na(Taxonomy_l_removed9_taxglom_edit)] <- "Unclassified"
# transform the table so that you add the aggregated abundances 
dc <- reshape2::dcast(Phylo.samples_i_rare_10_lefse, TaxonomyName~Sample, value.var = "Abundance", sum)

write.csv(dc, "/Users/LefseFile_Feces_PreChallenge.csv")
```

### Figure 6 and S9: Producing a heatmap of taxonomy names and correlation factors 
```{r}
Phylo.samples_i_filter_10 <- phyloseq_filter_prevalence(Phylo.samples_i_filter, prev.trh = 0.2, abund.trh = 10, abund.type = "mean", threshold_condition = "AND")
Phylo.samples_i_filter
Phylo.samples_i_filter_10

Taxonomy_i <- Phylo.samples_i_filter_10 %>%  
  transform_sample_counts(function(x) {x/sum(x)} ) %>%       # Transform to relative abundance
  psmelt()                                             # Melt to long format (this is for ggplot), create dataframe

Taxonomy_i$Abundance <- Taxonomy_i$Abundance*100

Taxonomy_i_edited <- Taxonomy_i %>%
  #filter(SampleType == "Duodenum" | SampleType == "Ileum") %>%
  #filter(SampleType == "Cecum") %>%
  #filter(SampleType == "Feces") %>%
  filter(Group_PostInfect != "Pre-Challenge")

# Change any NA to the name of the family
Taxonomy_i_edited$Genus <- ifelse(is.na(Taxonomy_i_edited$Genus), Taxonomy_i_edited$Family, Taxonomy_i_edited$Genus)

# Create a new column called TaxonomyName that combines the different taxonomic levels separated by ; 
Taxonomy_i_edited$TaxonomyName<-paste(Taxonomy_i_edited$Phylum, Taxonomy_i_edited$Genus,sep="| ")

write.csv(Taxonomy_i_edited, "/Users/hamster_Feces_heatmap_relativeabundances.csv")

# transform the table so that you add the aggregated abundances 
dc_otu <- dcast(Taxonomy_i_edited, SampleID~Genus, value.var = "Abundance", sum)

# Make the first column the row header 
rownames(dc_otu) <- dc_otu[,1]
# remove first column 
dc_otu <- dc_otu[,-1]
dc_otu_matrix <- data.matrix(dc_otu, rownames.force = TRUE)

# Perform log10 + 1 transformation
x = log10(dc_otu_matrix + 1)

#ALL SAMPLES
# import metadata for correlation analysis without hamster 9 
corr_metadata <- read.csv("/Users/SARS_hamster_metadata_correlation.csv")

#SMALL INTESTINE
corr_metadata <- read.csv("/Users/SARS_hamster_metadata_correlation_SI.csv")

#CECUM
corr_metadata <- read.csv("/Users/SARS_hamster_metadata_correlation_cecum.csv")

#FECES
corr_metadata <- read.csv("/Users/SARS_hamster_metadata_correlation_Feces.csv")

# Make the first column the row header 
rownames(corr_metadata) <- corr_metadata[,1]
# remove first column 
corr_metadata<- corr_metadata[,-1]
# since we turn into a matrix we set the groups to numbers (mock = 1, SARS = 2, IAV = 3) and (non-infected = 1 and infected = 2)

# correlate the two tables 

# Cross correlate data sets
correlation.table <- associate(x, corr_metadata, method = "spearman", mode = "table")
df <- correlation.table

df$X1 <- recode(df$X1, Bifidobacteriaceae = 'Unclassified Bifidobacteriaceae',
                           Muribaculaceae  = 'Unclassified Muribaculaceae',
                           Prevotellaceae = 'Unclassified Prevotellaceae',
                           Carnobacteriaceae  = 'Unclassified Carnobacteriaceae',
                           Erysipelotrichaceae = 'Unclassified Erysipelotrichaceae', 
                           Eubacteriaceae = 'Unclassified Eubacteriaceae',
                           Lachnospiraceae = 'Unclassified Lachnospiraceae',
                           Oscillospiraceae = 'Unclassified Oscillospiraceae',
                           Ruminococcaceae = 'Unclassified Ruminococcaceae',
                           Christensenellaceae = 'Unclassified Christensenellaceae',
                           Spirochaetaceae = 'Unclassified Spirochaetaceae')

levels(df$X1)[levels(df$X1)=='[Eubacterium] coprostanoligenes group'] <- 'Unclassified [Eubacterium] coprostanoligenes group'

df$X1 <- factor(df$X1, levels = c('Actinomyces', 'Bifidobacterium','Corynebacterium', 'Unclassified Bifidobacteriaceae', #Actinob
                                  
                                  'Alistipes', 'Bacteroides', 'Odoribacter','Parabacteroides', 'Prevotella', 
                                  'Prevotellaceae Ga6A1 group','Prevotellaceae UCG-001','Prevotellaceae UCG-003','Muribaculum',
                                  'Bacteroidales RF16 group', 'Unclassified Muribaculaceae', 'Unclassified Prevotellaceae', #Bacteroid
                                  
                                  'Campylobacter', 'Helicobacter', #Campy
                                  
                                  'Mucispirillum', #Deferri
                                  
                                  'Desulfovibrio', #Desulfo
                                  
                                  'Elusimicrobium', #Eluso
                                  
                                  'Fibrobacter', #Fibro
                                  
                                 'Allobaculum', 'Anaeroplasma', 'Faecalibaculum', 'Granulicatella', 'Ileibacterium', 'Ligilactobacillus', 'Mycoplasma','Streptococcus', 'Unclassified Erysipelotrichaceae', #Firmicutes baccili
                                  
                                  'Acetatifactor','Anaerostipes','Angelakisella','Christensenellaceae R-7 group', 'Colidextribacter', '[Eubacterium] siraeum group','Finegoldia','Fournierella', 'Incertae Sedis', 'Intestinimonas', 'Lachnoclostridium', 'Lachnospiraceae NK4A136 group', 'NK4A214 group', 'Oscillibacter', 'Peptococcus', 'Peptostreptococcus', 'Roseburia', 'Ruminococcus', '[Ruminococcus] torques group','Tuzzerella', 'UBA1819', 'UCG-003', 'UCG-005', 'UCG-010', 'Unclassified Christensenellaceae', 'Unclassified [Eubacterium] coprostanoligenes group', 'Unclassified Eubacteriaceae', 'Unclassified Lachnospiraceae', 'Unclassified Oscillospiraceae', 'Unclassified Ruminococcaceae', #Firmicutes clostridia
                                  
                                  'Megasphaera', 'Quinella', 'Veillonella', #Firmicutes negati
                                  
                                  'Fusobacterium', #Fusobacteria
                                
                                  'Coxiella','Haemophilus', 'Mannheimia', 'Parasutterella', #Proteobact
                                 
                                 'Treponema', 'Unclassified Spirochaetaceae'
                                  ))

df$X2 <- factor(df$X2, levels = c('SampleType','IAV', 'Challenged', 'R.NT', 'R.Trachea', 'R.Lung', 'R.Brain', 'R.Heart', 'R.Duodenum', 'R.Ileum', 'R.Cecum', 'P.NT', 'P.Trachea', 'P.Lung', 'P.Brain', 'P.Heart', 'P.SI', 'P.LI'))
 
df2 <- df %>%
  filter(X2 != "P.SI") %>%
  filter(X2 != "P.Heart")
                                  
theme_set(theme_bw(20))

# Plot using ggplot with legend
pdf("/Users/Correlation_Heatmap_GenusUnclassifiedFamilies_Feces_ExcludedPreChallenge.pdf",
    width = 9, height =12, 
    bg = NA) # Width and height in inches (width = 12, height =12 for all samples)

ggplot(df2, aes(X2, fct_rev(X1), group=X2)) +
    geom_tile(aes(fill = Correlation)) +
    geom_text(aes(label = round(Correlation, 2),
                  fontface = ifelse(Correlation > 0.60 | Correlation < -0.60, 2, 1), 
                  colour = ifelse(Correlation > 0.60 | Correlation < -0.60, "black", "white")),
                  size = 2.5) +
    scale_fill_gradientn("Correlation", 
                      breaks = seq(from = -1, to = 1,  by = 0.25), 
                      colours = c("blue", "white", "red"), 
                  limits = c(-1, 1)) +
    labs(x = "", y = "")+
    theme(axis.title.x=element_blank(),
          axis.title.y = element_blank(),
          axis.text.x = element_text(colour = "black", size = 9),
          axis.text.y = element_text(colour = "black", size = 10, face = "italic"),
          axis.line.x = element_line(color="black", size = 0.5),
          axis.line.y = element_line(color="black", size = 0.5),
          legend.title = element_blank(), 
          legend.text = element_text(size = 8)) +
  theme(panel.grid = element_blank(),
          axis.line = element_line(color="black", size = 0.5), 
          legend.position="bottom", 
          legend.key.width = unit(1.3, 'cm'))

dev.off()
```