---
title: "Sample Processing for Lung Hamster Samples"
output: html_notebook
---

# Before you begin:

These scripts were tailored for the analyses performed in:
  
Seibert et al, 2022, *Pathobiology and dysbiosis of the respiratory and intestinal microbiota in 14 months old Golden Syrian hamsters infected with SARS-CoV-2.*

# Purpose: 
This file will go through the initial sample processing and filtering to remove phyla that do not have an assigned Phyla, any Eukaryotes, chloroplasts and mitochondria, 

# Load the needed packages
```{r,message=FALSE}
library(phyloseq)
library(dplyr)
library(ggplot2)
library(metagMisc)
```

## Import files from dada2 scripts 

First, we will need to import the ASV count table, taxonomy file and the metadata file.\ 

*In order to read the metadata into a phyloseq object you need to make the SampleID (first column) the row headers*
```{r}
# Import metadata file (provided)
metadata <- read.csv("/path/Sars_hamster_metadata.csv", header=T, row.names=1, check.names=F)

# Import count table 
count_l <- read.table("/path/ASVs_counts.tsv", header=T, row.names=1, check.names=F, sep="\t")

# Import taxonomy file 
taxa_l <- as.matrix(read.table("/path/ASVs_taxonomy_dada_wSpecies.tsv", header=T, row.names=1, check.names=F, sep="\t"))
```

## Filtering variants for ONLY BACTERIA  

When the organisms from a set of samples are well-represented in the taxonomic reference database, it is advisable to filter variants in which a high-rank taxonomy could not be assigned. These are most likely sequence artifacts that donâ€™t exist in nature. Therefore, we will filter out any variants that are not classified as bacteria, have an unclassified phyla or are identified as Chloroplast or Mitochondria.\

### Import files into PhyloSeq
```{r}
# Import count table into phyloseq
ps_l = otu_table(count_l, taxa_are_rows = TRUE)

# Import taxonomy table into phyloseq 
tax_l = tax_table(taxa_l)

# Import metadata into phyloseq object
sample = sample_data(metadata)

# Merge the count table, taxonomy table and sample data 
Phylo_l = merge_phyloseq(ps_l, tax_l, sample)
Phylo_l
```

### First we will filter out any Phyla that are unassigned
```{r}
# Create table, number of features for each phyla
table(tax_table(Phylo_l)[, "Phylum"], exclude = NULL)

# Remove the ASVs and name as uncharacterized if the Phylum has an NA 
Phylo.filter_l <- subset_taxa(Phylo_l, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
table(tax_table(Phylo.filter_l)[, "Phylum"], exclude = NULL)
```

### Next we will filter out Eukaryotes, chloroplasts and mitochondria, because we only intended to amplify bacterial sequences
```{r}
Phylo.filter_l # Print information about the number of taxa and samples before filtering 
table(tax_table(Phylo.filter_l)[, "Kingdom"], exclude = NULL) # Print a table that states the number of sequences for the different kingdoms before filtering

Phylo.filter_filter_l <- Phylo.filter_l %>%
  subset_taxa(
    Kingdom == "Bacteria" &
    Class   != "Chloroplast" &
    Family  != "Mitochondria")

Phylo.filter_filter_l # Print information about the number of taxa and samples after filtering 
table(tax_table(Phylo.filter_filter_l)[, "Kingdom"], exclude = NULL) # Print a table that states the number of sequences for the different kingdoms after filtering
```

### Now we will separate the biological samples from the negative sequencing controls 
```{r}
# Subset the samples using column sampleClass in metadata
Phylo.samples_l <- Phylo.filter_filter_l %>%
  subset_samples(sampleClass == "sample")
Phylo.samples_l
```


### Next we will filter OTUs that are at low prevalence to try and remove sequencing artificacts. 

We will filter out any OTUs found in at least 5% of samples and total OTU abundance is greater that 10 reads 
```{r}
# Filter the samples with the specified conditions 
Phylo.samples_l_filter # Print information about the number of taxa and samples before filtering 
Phylo.samples_l_filter <- phyloseq_filter_prevalence(Phylo.samples_l, prev.trh = 0.05, abund.trh = 10, threshold_condition = "AND")
Phylo.samples_l_filter # Print information about the number of taxa and samples after filtering 
```

# Save the filtered phyloseq objects 
```{r}
# Save phyloseq files to a R file so that i can reload the filtered file 
saveRDS(object = Phylo.samples_l_filter, file = "/path/Phylo.samples_l.rds")
```
